

<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>PubChem3D: a new resource for scientists &mdash; Papers 1.0 documentation</title>
    
    <link rel="stylesheet" href="../_static/agogo.css" type="text/css" />
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '../',
        VERSION:     '1.0',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="../_static/jquery.js"></script>
    <script type="text/javascript" src="../_static/underscore.js"></script>
    <script type="text/javascript" src="../_static/doctools.js"></script>
    <link rel="top" title="Papers 1.0 documentation" href="../index.html" />
    <link rel="next" title="Open Babel: An open chemical toolbox" href="1758-2946-3-33.html" />
    <link rel="prev" title="Statistical filtering for NMR based structure generation" href="1758-2946-3-31.html" /> 
  </head>
  <body>
    <div class="header-wrapper">
      <div class="header">
        <div class="headertitle"><a
          href="../index.html">Papers 1.0 documentation</a></div>
        <div class="rel">
          <a href="1758-2946-3-31.html" title="Statistical filtering for NMR based structure generation"
             accesskey="P">previous</a> |
          <a href="1758-2946-3-33.html" title="Open Babel: An open chemical toolbox"
             accesskey="N">next</a> |
          <a href="../genindex.html" title="General Index"
             accesskey="I">index</a>
        </div>
       </div>
    </div>

    <div class="content-wrapper">
      <div class="content">
        <div class="document">
            
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body">
            
  <div class="section" id="pubchem3d-a-new-resource-for-scientists">
<h1>PubChem3D: a new resource for scientists<a class="headerlink" href="#pubchem3d-a-new-resource-for-scientists" title="Permalink to this headline">¶</a></h1>
<div class="section" id="abstract">
<h2>Abstract<a class="headerlink" href="#abstract" title="Permalink to this headline">¶</a></h2>
<div class="section" id="background">
<h3>Background<a class="headerlink" href="#background" title="Permalink to this headline">¶</a></h3>
<p>PubChem is an open repository for small molecules and their experimental biological activity. PubChem integrates and provides search, retrieval, visualization, analysis, and programmatic access tools in an effort to maximize the utility of contributed information. There are many diverse chemical structures with similar biological efficacies against targets available in PubChem that are difficult to interrelate using traditional 2-D similarity methods. A new layer called PubChem3D is added to PubChem to assist in this analysis.</p>
</div>
<div class="section" id="description">
<h3>Description<a class="headerlink" href="#description" title="Permalink to this headline">¶</a></h3>
<p>PubChem generates a 3-D conformer model description for 92.3% of all records in the PubChem Compound database (when considering the parent compound of salts). Each of these conformer models is sampled to remove redundancy, guaranteeing a minimum (non-hydrogen atom pair-wise) RMSD between conformers. A diverse conformer ordering gives a maximal description of the conformational diversity of a molecule when only a subset of available conformers is used. A pre-computed search per compound record gives immediate access to a set of 3-D similar compounds (called &#8220;Similar Conformers&#8221;) in PubChem and their respective superpositions. Systematic augmentation of PubChem resources to include a 3-D layer provides users with new capabilities to search, subset, visualize, analyze, and download data.</p>
<p>A series of retrospective studies help to demonstrate important connections between chemical structures and their biological function that are not obvious using 2-D similarity but are readily apparent by 3-D similarity.</p>
</div>
<div class="section" id="conclusions">
<h3>Conclusions<a class="headerlink" href="#conclusions" title="Permalink to this headline">¶</a></h3>
<p>The addition of PubChem3D to the existing contents of PubChem is a considerable achievement, given the scope, scale, and the fact that the resource is publicly accessible and free. With the ability to uncover latent structure-activity relationships of chemical structures, while complementing 2-D similarity analysis approaches, PubChem3D represents a new resource for scientists to exploit when exploring the biological annotations in PubChem.</p>
</div>
</div>
<div class="section" id="id1">
<h2>Background<a class="headerlink" href="#id1" title="Permalink to this headline">¶</a></h2>
<p>PubChem <a class="footnote-reference" href="#b1" id="id2">[1]</a><a class="footnote-reference" href="#b2" id="id3">[2]</a><a class="footnote-reference" href="#b3" id="id4">[3]</a><a class="footnote-reference" href="#b4" id="id5">[4]</a> (<a class="reference external" href="http://pubchem.ncbi.nlm.nih.gov">http://pubchem.ncbi.nlm.nih.gov</a>) is an open repository for small molecules and their experimental biological activities. The primary goal of PubChem is to be a public resource containing comprehensive information on the biological activities of small molecules. PubChem provides search, retrieval, visualization, analysis, and programmatic access tools in an effort to maximize the utility of contributed information. The PubChem3D project adds a new layer to this infrastructure. In the most basic sense, PubChem3D <a class="footnote-reference" href="#b5" id="id6">[5]</a><a class="footnote-reference" href="#b6" id="id7">[6]</a><a class="footnote-reference" href="#b7" id="id8">[7]</a><a class="footnote-reference" href="#b8" id="id9">[8]</a><a class="footnote-reference" href="#b9" id="id10">[9]</a><a class="footnote-reference" href="#b10" id="id11">[10]</a> generates a 3-D conformer model description of the small molecules contained within the PubChem Compound database. This 3-D description can be employed to enhance existing PubChem search and analysis methodologies by means of 3-D similarity. Prior to PubChem3D, this similarity approach was limited to a 2-D dictionary-based fingerprint (<a class="reference external" href="ftp://ftp.ncbi.nlm.nih.gov/pubchem/specifications/pubchem_fingerprints.txt">ftp://ftp.ncbi.nlm.nih.gov/pubchem/specifications/pubchem_fingerprints.txt</a>) to help relate chemical structures. With the advent of PubChem3D, this is now expanded to use a Gaussian-based similarity description of molecular shape <a class="footnote-reference" href="#b11" id="id12">[11]</a><a class="footnote-reference" href="#b12" id="id13">[12]</a><a class="footnote-reference" href="#b13" id="id14">[13]</a> used in software packages such as ROCS <a class="footnote-reference" href="#b14" id="id15">[14]</a> and OEShape <a class="footnote-reference" href="#b15" id="id16">[15]</a> from OpenEye Scientific Software, Inc.</p>
<p>It is reasonable to ask, why do we consider 3-D similarity methodologies at all? To put it simply, 2-D methods, while very useful and far cheaper computationally, may not be enough. A pitfall of most 2-D similarity methods is a general lack of ability to relate chemically diverse molecules with similar biological efficacy and function. For example, if a small molecule adopts an appropriate 3-D shape and possesses compatible functional groups properly oriented in 3-D space, it will likely bind to the biological moiety of interest. This &#8220;lock and key&#8221; binding motif is a major premise of structure-based drug design, docking, and molecular modelling applied with varying degrees of success over the past twenty years or more <a class="footnote-reference" href="#b16" id="id17">[16]</a><a class="footnote-reference" href="#b17" id="id18">[17]</a><a class="footnote-reference" href="#b18" id="id19">[18]</a><a class="footnote-reference" href="#b19" id="id20">[19]</a><a class="footnote-reference" href="#b20" id="id21">[20]</a><a class="footnote-reference" href="#b21" id="id22">[21]</a><a class="footnote-reference" href="#b22" id="id23">[22]</a><a class="footnote-reference" href="#b23" id="id24">[23]</a>. These &#8220;compatible functional groups&#8221; involved in binding small molecules to proteins, which are typically used to define pharmacophores, are referred to here simply as &#8220;features&#8221;. Therefore, in this context, 3-D similarity considering both shape and feature complementariness may be useful to find or relate chemical structures that may bind similarly to a protein target.</p>
<p>In its essence, 3-D similarity adds another dimension to data mining and it can provide some degree of orthogonality from 2-D similarity results. With 2-D similarity, one can typically see by eye increased changes in the chemical structure molecular graph with increasing dissimilarity <a class="footnote-reference" href="#b8" id="id25">[8]</a><a class="footnote-reference" href="#b10" id="id26">[10]</a>. With 3-D similarity, it is not always obvious by looking only at the molecular graph, often requiring one to visualize 3-D conformer alignments to relate diverse chemistries. In all, 3-D similarity is complementary to 2-D similarity and provides an easy-to-grasp understanding (<em>i.e.</em>, one can readily see by examining a conformer pair superposition that both shape and features are similar) that may help to provide a contrast or new insight to the same (biological) data.</p>
<p>This work gives an overview of the PubChem3D project and its current capabilities. The technology and background that allowed 3-D methodologies to be economically applied to the tens of millions of chemical structures in the PubChem Compound database are described elsewhere <a class="footnote-reference" href="#b5" id="id27">[5]</a><a class="footnote-reference" href="#b6" id="id28">[6]</a><a class="footnote-reference" href="#b7" id="id29">[7]</a><a class="footnote-reference" href="#b8" id="id30">[8]</a><a class="footnote-reference" href="#b9" id="id31">[9]</a><a class="footnote-reference" href="#b10" id="id32">[10]</a> covering various aspects of the project, including conformer model generation validation <a class="footnote-reference" href="#b6" id="id33">[6]</a>, the relative uniqueness of molecular shape <a class="footnote-reference" href="#b7" id="id34">[7]</a>, and 3-D neighboring methodology <a class="footnote-reference" href="#b8" id="id35">[8]</a>.</p>
</div>
<div class="section" id="construction-and-content">
<h2>Construction and Content<a class="headerlink" href="#construction-and-content" title="Permalink to this headline">¶</a></h2>
<div class="section" id="pubchem3d-coverage">
<h3>1. PubChem3D Coverage<a class="headerlink" href="#pubchem3d-coverage" title="Permalink to this headline">¶</a></h3>
<p>As one can imagine, it does not make sense nor is it possible to compute a 3-D description for all chemical structures in PubChem (<em>e.g.</em>, complexes and mixtures). PubChem provides a 3-D conformer model description for each record in the PubChem Compound database that satisfies the following conditions:</p>
<ol class="arabic simple">
<li>Not too large (with ≤ 50 non-hydrogen atoms).</li>
<li>Not too flexible (with ≤ 15 rotatable bonds).</li>
<li>Consists of only supported elements (H, C, N, O, F, Si, P, S, Cl, Br, and I).</li>
<li>Has only a single covalent unit (<em>i.e.</em>, not a salt or a mixture).</li>
<li>Contains only atom types recognized by the MMFF94s force field <a class="footnote-reference" href="#b24" id="id36">[24]</a><a class="footnote-reference" href="#b25" id="id37">[25]</a><a class="footnote-reference" href="#b26" id="id38">[26]</a>.</li>
<li>Has fewer than six undefined atom or bond stereo centers.</li>
</ol>
<p>Figure <a class="reference internal" href="1758-2946-3-46.html#figure-1"><em>1</em></a> shows the PubChem3D coverage as of June 2011. Out of more than 30.3 million chemical structure records in the PubChem Compound database, there are nearly 27.2 million records with a 3-D description. This represents 89.6% of the PubChem Compound contents (92.3% when considering that 2.7% are salts whose parent structure has a 3-D description). Of the remaining 7.7% of chemical structures in PubChem devoid of a 3-D description, the largest category (representing 1.48 million or 4.9% of the total archive) consists of structures with more than 15 rotatable bonds. The next largest unique count (<em>i.e.</em>, those not already represented by structures with more than 15 rotatable bonds) is the cases of MMFF94s non-supported elements and non-supported atom environments (representing 280 thousand or 0.9% of the total archive, with an overlapping absolute count of 389 thousand). The remaining unique counts are the cases of large structures with +50 non-hydrogen atoms (representing 253 thousand or 0.8% of the total archive, with an overlapping absolute count of 882 thousand), excessive undefined stereo (representing 129 thousand or 0.4% of the total archive, with an overlapping absolute count of 234 thousand), chemical structures involving complexes or mixtures (representing 105 thousand or 0.3% of the total archive, with an overlapping absolute count of 324 thousand), and conformer generation failure (representing 79 thousand or 0.3% of the total archive). While the reasons for missing a 3-D description categories sometimes overlap, the ordering above is such that the one with the largest overall population is chosen first, with each subsequent category picking the largest remaining unique subpopulation not already covered, until all categories were exhausted.</p>
<div class="figure align-center" id="figure-1">
<img alt="../_images/1758-2946-3-32-1.jpg" src="../_images/1758-2946-3-32-1.jpg" />
<p class="caption">Figure 1. PubChem Compound database 3-D coverage</p>
<div class="legend">
<strong>PubChem Compound database 3-D coverage</strong>. As one can see, 89.6% of all records have a 3-D conformer model. If one includes the parent compound of salts, this coverage can be considered to be 92.3%. Of the cases not having a 3-D conformer model, the majority are due to the flexibility of the chemical structure being too great to be suitable for conformer generation.</div>
</div>
</div>
<div class="section" id="conformer-models">
<h3>2. Conformer Models<a class="headerlink" href="#conformer-models" title="Permalink to this headline">¶</a></h3>
<p>The computed coordinates for the 3-D representations are the essence of the PubChem3D project. Creation of the stored conformational models consists of multistep processes involving separate conformer generation, sampling, and post processing steps.</p>
<p>All conformers were generated by the OpenEye Scientific Software, Inc., OMEGA software <a class="footnote-reference" href="#b27" id="id39">[27]</a><a class="footnote-reference" href="#b28" id="id40">[28]</a><a class="footnote-reference" href="#b29" id="id41">[29]</a><a class="footnote-reference" href="#b30" id="id42">[30]</a><a class="footnote-reference" href="#b31" id="id43">[31]</a> using the C++ interface, the MMFF94s force field <a class="footnote-reference" href="#b24" id="id44">[24]</a><a class="footnote-reference" href="#b25" id="id45">[25]</a><a class="footnote-reference" href="#b26" id="id46">[26]</a> minus coulombic terms, and an energy filter of 25 kcal/mol. (Removal of coulombic terms <a class="footnote-reference" href="#b6" id="id47">[6]</a><a class="footnote-reference" href="#b32" id="id48">[32]</a><a class="footnote-reference" href="#b33" id="id49">[33]</a><a class="footnote-reference" href="#b34" id="id50">[34]</a><a class="footnote-reference" href="#b35" id="id51">[35]</a> eliminated a bias towards conformations with energy-lowering intra-molecular interactions that tend not to be important for inter-molecular interactions, an important consideration given that the 3-D coordinates are generated in vacuo. Removal of attractive van der Waals terms did not have any noticeable effect <a class="footnote-reference" href="#b6" id="id52">[6]</a>.) A maximum of 100,000 conformers per chemical structure stereo isomer were allowed. When undefined stereo centers were present, each stereo isomer was enumerated and conformers independently generated. These stereo isomer conformers were then combined (2**5 = 32 maximum stereo permutations, 32 * 100,000 = maximum 3.2 million conformers).</p>
<p>Limiting to 100,000 conformations per stereo isomer can be a significant factor in limiting exploration of the conformational space. Ideally, one would want to explore the conformational space of a molecule exhaustively. In reality, it is not tractable to do so. For example, if one considers only three angles per rotatable bond and there are eleven rotatable bonds, this would yield 3**11 (= 177,147) possible conformers. If one considers four torsion angles per rotatable bond, and there are nine rotatable bonds, this would yield 4**9 (= 262,144) possible conformers. One can see how quickly systematic approaches can run into trouble with such exponential growth in the count of conformations and why there is a limit on how flexible a molecule is allowed to be.</p>
<p>With conformers generated, another important consideration is immediately obvious. It is not practical to store many thousands of conformers per compound. Therefore, after conformer generation is complete, the conformation count is reduced by sampling using root-mean-square-distance (RMSD) of pair-wise comparison of non-hydrogen atomic coordinates using the OEChem <a class="footnote-reference" href="#b36" id="id53">[36]</a> OERMSD function with the automorph detection (which considers local symmetry equivalence of atoms such that, for example, rotation of a phenyl ring does not yield an artificially high RMSD) and overlay (which minimizes RMSD between conformers by rotation and translation of one conformer to the other) options selected. In some rare cases, the automorph detection was prohibitively expensive computationally and not used.</p>
<p>The sampling procedure employed is described elsewhere <a class="footnote-reference" href="#b7" id="id54">[7]</a> but involves a two-stage clustering approach with an initial pass to partition-cluster conformers using an exclusion region hierarchy of decreasing dissimilarity (NlogN computational complexity, each cluster representative forms an exclusion region at a particular RMSD), followed by a step to remove edge-effects from the partition clustering (N<sup>2</sup> computational complexity using only the cluster representatives at the desired RMSD). The RMSD value used when sampling was dependent on the size and flexibility of the chemical structure.</p>
<p><strong>Equations 1</strong> and <strong>2</strong> were developed <a class="footnote-reference" href="#b6" id="id55">[6]</a> to help prevent using a conformer sampling RMSD that was less than the capability of the OMEGA software to reproduce bioactive ligand conformations. The equations were intended to ensure that 90% of the sampled conformer models of 25,972 small-molecule ligands, whose 3-D structures were experimentally determined, should contain at least one conformer within the RMSD sampling value to a bioactive conformation. The resulting <em>RMSD_pred</em> value was rounded to the nearest 0.2 increment. The smallest RMSD value used was 0.4. If more than 500 conformers resulted after sampling, the RMSD was incremented by a further 0.2 and the conformer model was re-clustered. This process was repeated as many times as necessary to restrict the overall count of conformers to be 500 or less.</p>
<p>where &#8220;<em>nha</em>&#8221; is the count of non-hydrogen atoms in the molecule, &#8220;<em>er</em>&#8221; is the effective rotor count, and &#8220;<em>RMSD_pred</em>&#8221; is the predicted average accuracy for a given &#8220;<em>nha</em>&#8221; and &#8220;<em>er</em>&#8221; value.</p>
<p>where &#8220;<em>er</em>&#8221; is the effective rotor count, &#8220;<em>rb</em>&#8221; is the rotatable bond count (computed using the OEChem &#8220;IsRotor&#8221; function) and &#8220;<em>nara</em>&#8221; is the count of non-aromatic ring atom count (OEChem OpenEye aromaticity model) excluding bridgehead atoms and SP2 hybridized atoms.</p>
<p>A post processing step was performed, after conformer model RMSD sampling, to completely relax the hydrogen atom locations by performing a full energy minimization where all non-hydrogen atoms were kept frozen. A subsequent &#8220;bump&#8221; check removed any conformers that had MMFF94 atom-atom interactions greater than 25 kcal/mol. Finally, each conformer was rotated and translated to their principal steric axes (<em>i.e.</em>, non-mass weighted principal moments of inertia axes) considering only non-hydrogen atoms.</p>
<p>It is important to note that the conformers produced are not stationary points on a potential energy hypersurface. In fact, one can readily achieve lower-energy conformations of a given chemical structure by performing an all-atom energy minimization to remove any bond, angle, or torsion strain present in vacuo. The PubChem3D conformer model for a chemical structure is meant to represent all possible biologically-relevant conformations that the molecule may have. In theory, one should have a reasonable chance to find any biologically accessible conformation within the RMSD sampling distance of the conformer model.</p>
</div>
<div class="section" id="conformer-model-properties">
<h3>3. Conformer Model Properties<a class="headerlink" href="#conformer-model-properties" title="Permalink to this headline">¶</a></h3>
<p>After a conformer model is produced, a series of properties are computed for each compound and each associated conformer. Table <em class="xref std std-ref">1</em> lists the compound- and conformer-level properties provided by PubChem3D. The compound properties include: the sampling RMSD used to construct the conformer model; the MMFF94 partial charges per atom <a class="footnote-reference" href="#b36" id="id56">[36]</a>; the functional group atoms that define each pharmacophore feature <a class="footnote-reference" href="#b15" id="id57">[15]</a>; and the diverse conformer ordering, always starting with the default conformer per compound.</p>
<p>The feature definition lists the set of non-hydrogen atoms that comprise a given fictitious feature atom. The feature definitions are computed using the OEShape &#8220;ImplicitMillsDeans&#8221; forcefield <a class="footnote-reference" href="#b15" id="id58">[15]</a><a class="footnote-reference" href="#b37" id="id59">[37]</a>. Care is taken to (iteratively) merge feature definitions of common type that are within 1.0 Å distance of each other. Each feature definition is used to generate a fictitious &#8220;color&#8221; atom, whose 3-D coordinates are at the steric center of the atoms that comprise it (<em>i.e.</em>, at the average {X, Y, Z} value). There are six feature types used: anion, cation, (hydrogen-bond) acceptor, (hydrogen-bond) donor, hydrophobe, and ring.</p>
<p>The conformer properties include: the global conformer identifier (GID); conformer volume <a class="footnote-reference" href="#b15" id="id60">[15]</a>; steric shape moments (monopole, quadrupole {Q<sub>x</sub>, Q<sub>y</sub>, Q<sub>z</sub>}, and octopole {O<sub>xxx</sub>, O<sub>yyy</sub>, O<sub>zzz</sub>, O<sub>xxy</sub>, O<sub>xxz</sub>, O<sub>yyx</sub>, O<sub>yyz</sub>, O<sub>zzx</sub>, O<sub>zzy</sub>, and O<sub>xyz</sub>}) <a class="footnote-reference" href="#b15" id="id61">[15]</a>; shape self-overlap volume used in shape similarity computations <a class="footnote-reference" href="#b11" id="id62">[11]</a>; feature self-overlap volume used in feature similarity computations <a class="footnote-reference" href="#b11" id="id63">[11]</a>; MMFF94s energy with coulombic terms removed <a class="footnote-reference" href="#b38" id="id64">[38]</a>; and the PubChem shape fingerprint <a class="footnote-reference" href="#b8" id="id65">[8]</a>.</p>
<p>where <em>ST</em> is the measure of shape similarity (shape Tanimoto), <sub>AA</sub> Vand <sub>BB</sub> Vare the respective self-overlap volume of conformers A and B, and <sub>AB</sub> Vis the common overlap volume between them.</p>
<p>where <em>CT</em> is the measure of feature similarity (color Tanimoto), the index &#8220;<em>f</em>&#8221; indicates any of the six independent fictitious feature atom types, <em>f</em>, and <em>f</em>.</p>
<p>where <em>ComboT</em> is the combo Tanimoto, <em>ST</em> is the shape Tanimoto, and <em>CT</em> is the color Tanimoto.</p>
<p>A diverse ordering of conformers is provided for each compound conformer ensemble <a class="footnote-reference" href="#b8" id="id66">[8]</a><a class="footnote-reference" href="#b39" id="id67">[39]</a><a class="footnote-reference" href="#b40" id="id68">[40]</a>. Using the lowest energy conformer in the ensemble as the initial default conformer, the conformer most dissimilar to the first is selected as the second diverse conformer. The conformer most dissimilar to the first two dissimilar conformers is chosen as the third diverse conformer. This process is repeated until there are no more conformers to be assigned a dissimilarity ordering. Similarity is measured by ST (<strong>Equation 3</strong>) and CT (<strong>Equation 4</strong>), involving a conformer superposition optimization <a class="footnote-reference" href="#b11" id="id69">[11]</a><a class="footnote-reference" href="#b36" id="id70">[36]</a> to maximize the shape volume overlap between two conformers by means of rotation and translation of one conformer to the other. This is followed by a single point CT computation at the ST-optimized conformer pair overlay. The ST and CT are then added to yield a combo Tanimoto (<strong>Equation 5</strong>). The conformer with the smallest sum of combo Tanimoto to all <em>assigned</em> dissimilar conformers is selected as the next most dissimilar. In the case of a tie, the one with the largest sum of combo Tanimoto to <em>unassigned</em> conformers is used.</p>
<p>Note that PubChem has another source of 3-D information of small molecules, besides PubChem3D. The PubChem Substance database (unique identifier: SID) contains 3-D structures of small molecules deposited from individual depositors, which can be either experimentally determined or computationally predicted. For clarification, these depositor-provided structures are called &#8220;<em>substance</em> conformers&#8221;, and the theoretical conformers generated by PubChem3D for each PubChem Compound record (unique identifier: CID) are called &#8220;<em>compound</em> conformers&#8221;. For an efficient use of the PubChem3D resources, it is necessary to assign a unique identifier to each of compound conformers in the PubChem Compound database and substance conformers in the PubChem Substance database. The global conformer identifier (GID) uniquely identifies each conformer and is stored as a hex-encoded 64-bit unsigned integer, where the first 16-bits (0x000000000000FFFF) correspond to the local conformer identifier (LID), which is specific to a given conformer ensemble, the next 16-bits (0x00000000FFFF0000) are the version identifier (always zero for PubChem3D compound conformers, but nonzero for deposited substance conformers) and the last 32 bits (0xFFFFFFFF00000000) correspond to the structure identifier. This identifier is a compound identifier (CID), if the version identifier is zero, and a substance identifier (SID), when the version identifier is non-zero (the version identifier indicates the substance version to which the conformer corresponds). Substance conformer identifiers allow deposited 3-D coordinates to be utilized effectively by the PubChem3D system. As one can see, the GID provides global conformer identification system across all PubChem conformers.</p>
<p>A shape fingerprint is computed for the first ten diverse conformers. To generate this property, each conformer is ST-optimized to a set of reference conformers that describe the entire shape space diversity of the contents of PubChem3D. If the conformer is shape similar beyond a particular threshold to a reference conformer, the reference conformer identifier (CID and LID) and a packed rotational/translational matrix (64-bit integer) are retained. This makes each set reference conformer like a bit in a binary fingerprint, however; in this case, additional information (the superposition) is also retained. One can imagine that these shape fingerprints are a little like coordinates in shape space, mapping where a given conformer is located.</p>
<p>This shape fingerprint can be used in several ways during 3-D similarity computation and was born out of our earlier research <a class="footnote-reference" href="#b8" id="id71">[8]</a><a class="footnote-reference" href="#b41" id="id72">[41]</a> on &#8220;alignment recycling.&#8221; This work demonstrated that similar conformers align to a reference shape in a similar way. This means that, if one is interested only in finding similar shapes, conformer pairs that do not have common shape fingerprint &#8220;bits&#8221; can be ignored (<em>i.e.</em>, there is no need to perform a computationally intensive conformer alignment overlap optimization between two conformers when no common shape fingerprint reference exists, because the two conformer shapes are dissimilar to the extent that they may not need to be considered further). Additionally, when a common shape fingerprint reference exists between two conformers, one can &#8220;replay&#8221; the alignments of the two conformers to the common reference shape to yield a conformer alignment overlap between conformers that is (typically) very close to the optimal overlay; thus speeding up any conformer alignment overlap optimization but also providing an opportunity to further skip overlap optimization, when the best pre-optimized alignment overlap is not sufficient.</p>
</div>
<div class="section" id="similar-conformer-neighboring-relationship">
<h3>4. Similar Conformer Neighboring Relationship<a class="headerlink" href="#similar-conformer-neighboring-relationship" title="Permalink to this headline">¶</a></h3>
<p>Analogous to the precomputed &#8220;Similar Compounds&#8221; relationship for 2-D similarity, PubChem3D now provides a &#8220;Similar Conformers&#8221; neighboring relationship <a class="footnote-reference" href="#b8" id="id73">[8]</a> using 3-D similarity. This neighboring takes into account both conformer shape similarity and conformer pharmacophore feature similarity. Essentially, this is equivalent to performing a shape-optimized similarity search using ROCS <a class="footnote-reference" href="#b14" id="id74">[14]</a><a class="footnote-reference" href="#b15" id="id75">[15]</a> at a threshold of ST &gt; 0.795 and CT &gt; 0.495, when both conformers have defined pharmacophore features. To allow for compounds devoid of features to be neighbored, a threshold of ST &gt; 0.925 is used, but with the caveat that both conformers must not have any defined pharmacophore features. Currently, three diverse conformers per compound are neighbored; however, this may change, with up to ten conformers per compound used as computational resources allow. The conformers used for neighboring correspond to the first &#8220;N&#8221; conformers in the diverse conformer list property. (See the <strong>Conformer Model Properties</strong> section.) This ensures maximal coverage of the unique shape/feature space of a chemical structure as additional conformers are considered in neighboring.</p>
</div>
<div class="section" id="ftp-site">
<h3>5. FTP Site<a class="headerlink" href="#ftp-site" title="Permalink to this headline">¶</a></h3>
<p>PubChem3D data is available on the PubChem FTP site (<a class="reference external" href="ftp://ftp.ncbi.nlm.nih.gov/pubchem/Compound_3D">ftp://ftp.ncbi.nlm.nih.gov/pubchem/Compound_3D</a>). One may download in bulk 3-D descriptions of PubChem Compound records. On average there are approximately 110 conformers per compound in the PubChem3D system; however, not all data is provided for public download, in part due to the overall size being many terabytes, more data than one can readily share publicly. Therefore, two different subsets are provided in various file formats (SDF, XML, and ASN.1) that correspond to either the default conformer or the first ten conformers in the diverse conformer list property. (See the <strong>Conformer Model Properties</strong> section.) Beyond these two conformer subsets of PubChem3D, one may also find a description of the conformers that comprise the PubChem3D shape fingerprint. These conformers represent all shape diversity present in the PubChem3D system for a given analytic volume range and a given level of shape similarity ST threshold.</p>
<p>The &#8220;Similar Conformers&#8221; neighboring relationship is also provided for download. This conformer pair relationship (one per line) includes the respective conformer identifiers, ST, CT, and the 3 × 3 rotation matrix and translation vector (applied in that order) to superimpose the second conformer to the first. The rotation/translation refers to the coordinates provided in the download set of ten diverse conformers or otherwise available for download from our PubChem download facility. (See the <strong>Utility: Download</strong> section.)</p>
</div>
</div>
<div class="section" id="utility">
<h2>Utility<a class="headerlink" href="#utility" title="Permalink to this headline">¶</a></h2>
<div class="section" id="ncbi-entrez-interface">
<h3>1. NCBI Entrez Interface<a class="headerlink" href="#ncbi-entrez-interface" title="Permalink to this headline">¶</a></h3>
<p>The primary search interface for PubChem is Entrez <a class="footnote-reference" href="#b4" id="id76">[4]</a>, <em>e.g.</em>, for the PubChem Compound database, accessible by means of the PubChem homepage (<a class="reference external" href="http://pubchem.ncbi.nlm.nih.gov">http://pubchem.ncbi.nlm.nih.gov</a>) or the URL: <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pccompound?Db=pccompound">http://www.ncbi.nlm.nih.gov/pccompound?Db=pccompound</a>. There are fourteen Entrez indexes available to query PubChem Compound records based on 3-D information detailed in Table <em class="xref std std-ref">2</em>. For example, to find which compound conformer models were sampled in the RMSD range between 0.4 and 0.6, one would perform the query</p>
<p>The indexes for &#8220;Volume3D&#8221;, &#8220;XStericQuadrupole3D&#8221;, &#8220;YStericQuadrupole3D&#8221;, and &#8220;ZStericQuadrupole3D&#8221; correspond, respectively, to the analytic volume and the three steric quadrupole moments <a class="footnote-reference" href="#b9" id="id78">[9]</a><a class="footnote-reference" href="#b12" id="id79">[12]</a><a class="footnote-reference" href="#b42" id="id80">[42]</a> for only the first conformer in the diverse conformer list (<em>i.e.</em>, the default conformer). The steric quadrupoles essentially correspond to the extents of the compound, where X, Y, and Z correspond to the length, width, and height. For example, to find very long, near-linear compounds, one may give the PubChem Compound Entrez query <a class="reference external" href="http://pubchem.ncbi.nlm.nih.gov/help.html#PubChem_index">http://pubchem.ncbi.nlm.nih.gov/help.html#PubChem_index</a>).</p>
<p>PubChem also provides filtering capabilities. Unlike indexes, which hold discrete values, filters are Boolean-based (<em>i.e.</em>, either a record is in the list or it is not). PubChem3D provides some additional filtering capabilities. In the case of the PubChem Compound database, there is a filter &#8220;has 3d conformer&#8221; that will indicate whether a given compound record has a 3-D conformer model by means of the PubChem Compound query: &#8220;</p>
<p>Filtering capabilities were also expanded in the PubChem Substance database. Two filters were added: &#8220;has deposited 3d&#8221; and &#8220;has deposited 3d experimental&#8221; to indicate when a substance record has 3-D coordinates and when the contributed 3-D coordinates were determined experimentally, respectively. For example, to find all experimentally determined 3-D structures for substance records, one would use the PubChem Substance databases query: &#8220;</p>
</div>
<div class="section" id="visualization">
<h3>2. Visualization<a class="headerlink" href="#visualization" title="Permalink to this headline">¶</a></h3>
<p>Each PubChem Compound (and Substance) record has a summary page as depicted in Figure <a class="reference internal" href="1758-2946-3-46.html#figure-2"><em>2</em></a> (<a class="reference external" href="http://pubchem.ncbi.nlm.nih.gov/summary/summary.cgi?cid=681">http://pubchem.ncbi.nlm.nih.gov/summary/summary.cgi?cid=681</a> for dopamine). When a 3-D conformer model can be produced for a compound record (or a depositor-provided 3-D coordinates for the substance record), a 3-D image of the structure will be available by clicking the &#8220;3D&#8221; tab. In the case of a PubChem Compound record, this corresponds to the first diverse conformer, which is the default conformer. As shown in Figure <a class="reference internal" href="1758-2946-3-46.html#figure-3"><em>3</em></a>, if one clicks the image, a popup menu appears allowing one to invoke the &#8220;Web-based 3D Viewer&#8221; or to send the 3-D information to the &#8220;Pc3D Viewer Application&#8221;.</p>
<div class="figure align-center" id="figure-2">
<img alt="../_images/1758-2946-3-32-2.jpg" src="../_images/1758-2946-3-32-2.jpg" />
<p class="caption">Figure 2. Summary page enhancements</p>
<div class="legend">
<strong>Summary page enhancements</strong>. A snapshot of the PubChem Compound summary page of dopamine (CID 681). Clicking on the &#8220;3D&#8221; tab on the right side of the page shows the 3-D structure of the molecule. Clicking the &#8220;Compound information&#8221; in the &#8220;Table of Contents&#8221; box directs users to 2-D neighbors (&#8220;Similar Compounds&#8221;) and 3-D neighbors (&#8220;Similar Conformers&#8221;).</div>
</div>
<div class="figure align-center" id="figure-3">
<img alt="../_images/1758-2946-3-32-3.jpg" src="../_images/1758-2946-3-32-3.jpg" />
<p class="caption">Figure 3. Visualization of a 3-D structure conformer</p>
<div class="legend">
<strong>Visualization of a 3-D structure conformer</strong>. Clicking on the 3-D image on the PubChem Compound summary page (left) shows links to the web-based 3-D viewer (top right) and the Pc3D desktop helper application (bottom right).</div>
</div>
<p>The Pc3D viewer application can be downloaded and installed on PC, Mac, or Linux computers. A link to download this application can be found below the image on a given summary page or other PubChem3D aware pages (<em>e.g.</em>, see the &#8220;Pc3D Viewer Download&#8221; icon in Figure <a class="reference internal" href="1758-2946-3-46.html#figure-2"><em>2</em></a>). The viewer provides an interface for rendering 3-D structures of PubChem Compound records and visualizing their superpositions. With a customizable 3-D rendering engine that provides dynamic molecular visualization experience, it has the ability to create high-resolution, publication-quality images. It allows use of XYZ model files and SDF files and supports PubChem native formatted files (with the .pc3d or .asn extension).</p>
<p>The web-based 3-D viewer, like the Pc3D viewer application, allows one to browse 3-D conformers available for substances or compounds and their superpositions. This interactive tool (accessible via <a class="reference external" href="http://pubchem.ncbi.nlm.nih.gov/vw3d/">http://pubchem.ncbi.nlm.nih.gov/vw3d/</a>) operates without the need for a web browser plug-in (and does not use Java, for support related reasons) by means of displaying a series of images to simulate molecule rotation. As shown in Figure <a class="reference internal" href="1758-2946-3-46.html#figure-4"><em>4</em></a>, besides providing immediate access to the &#8220;Similar Conformer&#8221; neighboring relationship per compound (and per compound conformer), users can access various controls to perform such tasks as: superposition or conformer navigation, data export, conformer rotation type, conformer rotation speed, conformer image resize, conformer filtering, and sorting. The viewer allows any arbitrary set of 3-D compound conformers or conformer pairs (substance and compound) that exists within PubChem to be viewed or superimposed. This tool is also the primary resource to visualize and manage 3-D information from various PubChem3D-aware tools, including 3-D conformer search and 3-D structure clustering.</p>
<div class="figure align-center" id="figure-4">
<img alt="../_images/1758-2946-3-32-4.jpg" src="../_images/1758-2946-3-32-4.jpg" />
<p class="caption">Figure 4. Visualization of 3-D structure conformer superpositions</p>
<div class="legend">
<strong>Visualization of 3-D structure conformer superpositions</strong>. Superpositions between compound conformers are accessible from various PubChem3D-aware applications. The PubChem Compound summary page (top left) allows the &#8220;Similar Conformers&#8221; neighboring relationship to be visualized. The PubChem3D web-based viewer (bottom left) allows arbitrary superpositions to be generated. The PubChem Structure Clustering tool (bottom right) enables all pair-wise superpositions to be examined.</div>
</div>
</div>
<div class="section" id="search">
<h3>3. Search<a class="headerlink" href="#search" title="Permalink to this headline">¶</a></h3>
<p>The PubChem Structure Search system <a class="footnote-reference" href="#b1" id="id81">[1]</a> (accessible via <a class="reference external" href="http://pubchem.ncbi.nlm.nih.gov/search/">http://pubchem.ncbi.nlm.nih.gov/search/</a>) allows one to search the PubChem Compound database using a chemical structure in various formats. PubChem3D adds a new capability to this system by allowing one to perform a 3-D similarity search and to visualize the results. At the time of writing, this similarity search is essentially equivalent to that described in the <strong>Similar Conformer Neighboring Relationship</strong> section. If 3-D coordinates are not provided for a chemical structure query, they are generated automatically, as-is possible, while keeping in mind that not all chemical structures can be covered by the PubChem3D system. (See the <strong>PubChem3D Coverage</strong> section for more details.) To aid in performing automated queries, a programmatic interface is available. (See the <strong>Programmatic Interface</strong> section for more details.)</p>
<p>A 3-D conformer search currently considers the first three diverse conformers per compound as candidates for &#8220;Similar Conformers&#8221;. (See diverse conformer ordering in the <strong>Conformer Model Properties</strong> section.) Given that there are more than 27 million CIDs and three conformers per compound are being considered, this means that there are around 81 million conformers considered by each 3-D query. This count will change as a function of time as data is added to PubChem and as the count of conformers per compound are increased. To achieve adequate query throughput, an &#8220;embarrassingly parallel divide-and-conquer&#8221; strategy is employed. The PubChem Compound conformer data set is subdivided into multiple evenly-sized subsets. Each subset is then searched in parallel. If more query throughput is desired, and the computational capacity exists, the solution is simple; one simply needs to increase the count of evenly-sized subsets to simultaneously process.</p>
</div>
<div class="section" id="download">
<h3>4. Download<a class="headerlink" href="#download" title="Permalink to this headline">¶</a></h3>
<p>The PubChem Download facility <a class="footnote-reference" href="#b1" id="id82">[1]</a> (<a class="reference external" href="http://pubchem.ncbi.nlm.nih.gov/pc_fetch">http://pubchem.ncbi.nlm.nih.gov/pc_fetch</a>) allows one to download PubChem records resulting from a search or a user-provided identifier list. With the advent of the PubChem3D layer, there is now the ability to download up to ten diverse conformers per compound. Alternatively, 3-D images may be downloaded (for the default conformer, only). A programmatic interface is available. (See the <strong>Programmatic Interface</strong> section for more details.)</p>
</div>
<div class="section" id="similarity-computation">
<h3>5. Similarity Computation<a class="headerlink" href="#similarity-computation" title="Permalink to this headline">¶</a></h3>
<p>The PubChem Score Matrix facility (<a class="reference external" href="http://pubchem.ncbi.nlm.nih.gov/score_matrix">http://pubchem.ncbi.nlm.nih.gov/score_matrix</a>) allows one to compute pairwise similarities of a set of PubChem compound records (up to 1,000,000 similarity pairs per request). The PubChem3D layer adds the ability to compute 3-D similarities using up to ten conformers (either the first <em>N</em>-diverse conformers or a user-provided conformer set) per compound per request. Additionally, this service allows one to select the type of superposition optimization (shape or feature) to perform. A programmatic interface is available. (See the <strong>Programmatic Interface</strong> section.)</p>
</div>
<div class="section" id="clustering-and-analysis">
<h3>6. Clustering and Analysis<a class="headerlink" href="#clustering-and-analysis" title="Permalink to this headline">¶</a></h3>
<p>The PubChem Structure Clustering tool <a class="footnote-reference" href="#b10" id="id83">[10]</a> (<a class="reference external" href="http://pubchem.ncbi.nlm.nih.gov/assay/assay.cgi?p=clustering">http://pubchem.ncbi.nlm.nih.gov/assay/assay.cgi?p=clustering</a>) allows one to perform single-linkage clustering for up to 4,000 compounds at a time. This interactive tool provides visualization, subset, selection, and analysis capabilities. For example, the dendrogram allows compounds to be grouped into clusters by clicking the Tanimoto bar provided above and below the dendrogram (see the bottom right panel in Figure <a class="reference internal" href="1758-2946-3-46.html#figure-4"><em>4</em></a>). One can then click on the cluster to view the individual compounds or perform other operations. The PubChem3D layer adds the ability to cluster compounds according to their 3-D similarities, with up to ten diverse conformers per compound. This service allows one to select: the superposition optimization type (shape or feature); whether to cluster all conformers or just the most similar conformer pair; and the conformer similarity metric.</p>
</div>
<div class="section" id="programmatic-interface">
<h3>7. Programmatic Interface<a class="headerlink" href="#programmatic-interface" title="Permalink to this headline">¶</a></h3>
<p>PubChem provides a programmatic interface called the Power User Gateway (PUG) <a class="footnote-reference" href="#b1" id="id84">[1]</a>. This extends the capabilities provided by the NCBI eUtils programmatic interface <a class="footnote-reference" href="#b43" id="id85">[43]</a>, which interfaces the NCBI Entrez search engine contents. PUG can be used to send programmatic requests (<em>e.g.</em>, to perform queries or other tasks). If a request does not complete, a request ID is returned. One uses this to &#8220;poll&#8221; whether the request is completed, at which point an URL is provided to obtain the results. This is necessary, considering that most user requests are queued and may not be executed or completed immediately. A PUG/SOAP interface exists to allow the SOAP-based protocol to be used to route requests. SOAP-interfaces are readily available for most programming (<em>e.g.</em>, Java, C#, VisualBasic) and scripting languages (<em>e.g.</em>, Perl, Python), as well as workflow applications (<em>e.g.</em>, Taverna <a class="footnote-reference" href="#b44" id="id86">[44]</a>, Pipeline Pilot <a class="footnote-reference" href="#b45" id="id87">[45]</a>). The PubChem3D layer extensions are now available in individual PUG-aware interfaces and by means of the PUG/SOAP interface.</p>
</div>
</div>
<div class="section" id="examples-of-use">
<h2>Examples of use<a class="headerlink" href="#examples-of-use" title="Permalink to this headline">¶</a></h2>
<p>To assist in understanding how PubChem3D can be useful to locate additional biological annotation and enhance one&#8217;s ability to identify potential structure-activity relationships, a series of illustrative examples were prepared. These examples benefit from a recent study <a class="footnote-reference" href="#b10" id="id88">[10]</a> of the statistical distribution of random 3-D similarities of more than 740,000 biologically tested small molecules in PubChem using a single conformer per compound, where the average (μ) and standard deviation (σ) of the shape-optimized ST, CT, and ComboT scores between two randomly selected conformers were found to be 0.54 ± 0.10, 0.07 ± 0.05, and 0.62 ± 0.13, respectively. The probability of two random conformers having a ST-optimized similarity score greater than or equal to the μ + 2σ threshold (<em>i.e.</em>, 0.74, 0.17, and 0.88 for ST, CT, and ComboT, respectively) was 2%, 4%, and 3% for ST, CT, and ComboT, respectively. This statistical information is meaningful to provide reasonable 3-D similarity thresholds, whereby one can be confident that most of the 3-D similarities between chemical structures is not simply by chance. When a group of chemical structures with similar biological activity and function are shown to have 3-D similarity to each other above these thresholds, it suggests that a common macromolecule binding interaction orientation exists and, furthermore, that the features required for such binding are present.</p>
<div class="section" id="finding-additional-biological-annotation">
<h3>1. Finding additional biological annotation<a class="headerlink" href="#finding-additional-biological-annotation" title="Permalink to this headline">¶</a></h3>
<p>In a data system such as PubChem, with a very uneven amount of biological annotation, it is helpful to find related chemical structures where more information is known. PubChem provides two precomputed neighboring relationships to locate similar chemical structures. The &#8220;Similar Conformers&#8221; neighboring relationship precomputes the 3-D similarity between all chemical structures in PubChem, while the &#8220;Similar Compounds&#8221; neighboring relationship precomputes the 2-D similarity. Using dopamine (CID 681) as an example, Figure <a class="reference internal" href="1758-2946-3-46.html#figure-5"><em>5</em></a> shows there can be relatively little commonality between 2-D and 3-D similarities; however, both relationships find chemicals that are related, with the 2-D similarity being good at finding chemical analogs of a given chemical while the 3-D similarity is skilled at locating molecules with similar shape and similar 3-D orientation of binding features. Therefore, use of both neighboring relationships allows a larger number of related chemicals to be found with associated biomedical literature (MeSH Links), biologically tested (BioAssay Tested), or bound to a protein 3-D structure (Protein3D Links).</p>
<div class="figure align-center" id="figure-5">
<img alt="../_images/1758-2946-3-32-5.jpg" src="../_images/1758-2946-3-32-5.jpg" />
<p class="caption">Figure 5. 3-D similarity relationship finds additional biological annotation</p>
<div class="legend">
<strong>3-D similarity relationship finds additional biological annotation</strong>. Comparison of the 2-D &#8220;Similar Compound&#8221; and 3-D &#8220;Similar Conformer&#8221; neighboring relationships using dopamine to demonstrate how both neighboring relationship complement each other when locating related chemical structures with unique biological annotation.</div>
</div>
</div>
<div class="section" id="relating-chemical-probes-for-same-biological-target">
<h3>2. Relating chemical probes for same biological target<a class="headerlink" href="#relating-chemical-probes-for-same-biological-target" title="Permalink to this headline">¶</a></h3>
<p>ML088 (CID 704205) and ML087 (CID 25199559), shown in Figure <a class="reference internal" href="1758-2946-3-45.html#figure-6"><em>6</em></a>, are chemical probes reported <a class="footnote-reference" href="#b46" id="id89">[46]</a> in a PubChem BioAssay (AID 1548) with EC50s of 6.19 μM and 0.20 μM, respectively. Both probes target a common protein, the tissue non-specific alkaline phosphatase (TNAP, GI 116734717), the deficiency of which is associated with defective bone mineralization in the form of rickets and osteomalacia. At first glance, these two chemical structures are rather dissimilar, with a 2-D subgraph similarity of 0.43 using the PubChem fingerprint. This suggests the two chemical structures are unrelated to each other, giving no hint as to why they have similar biological function and efficacy. Using 3-D similarity, by means of the PubChem3D web-based viewer as shown in Figure <a class="reference internal" href="1758-2946-3-45.html#figure-6"><em>6</em></a>, the shape, feature, and combo similarities (0.80, 0.23, and 1.03 for ST, CT, and ComboT, respectively) tell a very different story. The two chemical structures are 3-D similar, suggesting that the two chemical structures can adopt a similar shape and have some binding features in a common 3-D orientation, thus helping to relate the observed biological activity by providing a hypothesis that the two inhibitors may bind in a similar manner. While this could be interpreted as simply highlighting a deficiency in the PubChem 2-D similarity metric, in this case, PubChem 3-D similarity complements the PubChem 2-D similarity by allowing such a similarity relationship to be found between these two chemical probes.</p>
<div class="figure align-center" id="figure-6">
<img alt="../_images/1758-2946-3-32-6.jpg" src="../_images/1758-2946-3-32-6.jpg" />
<p class="caption">Figure 6. Relating biologically active compounds by means of PubChem3D</p>
<div class="legend">
<strong>Relating biologically active compounds by means of PubChem3D</strong>. Chemical probes ML088 (CID 704205) and ML087 (CID 25199559) from PubChem BioAssay 1548 against tissue non-specific alkaline phosphatase (TNAP, GI:116734717) are not similar by 2-D similarity but are by 3-D similarity.</div>
</div>
</div>
<div class="section" id="relating-chemically-diverse-structures-with-same-pharmacological-action">
<h3>3. Relating chemically diverse structures with same pharmacological action<a class="headerlink" href="#relating-chemically-diverse-structures-with-same-pharmacological-action" title="Permalink to this headline">¶</a></h3>
<p>Figure <a class="reference internal" href="1758-2946-3-45.html#figure-7"><em>7</em></a> shows the 2-D and 3-D similarity score matrices for a carefully selected set of eight anti-inflammatory drug molecules having the same MeSH <a class="footnote-reference" href="#b47" id="id90">[47]</a> pharmacological action annotation of &#8220;Histamine H1 Antagonists&#8221; (MeSH ID 68006634). Figure <a class="reference internal" href="1758-2946-3-45.html#figure-8"><em>8</em></a> depicts a subset of 3-D ST-optimized superpositions resulting from the 28 unique compound pairs. The 2-D Tanimoto similarity values between these compounds are quite low, with only three compound pairs above ≥ 0.75, indicating that the 2-D similarity method based on the PubChem fingerprint fails to interrelate their common biological activity as histamine H1 receptor antagonists. On the contrary, the 3-D similarity between these eight molecules is rather high, with a ST ≥ 0.74 and ComboT ≥ 1.0 for all but eight of the 28 compound pairs. As illustrated in Figure <a class="reference internal" href="1758-2946-3-45.html#figure-8"><em>8</em></a>, even if the 2-D Tanimoto value between a pair of molecules is as low as 0.31, they can still have significant structural overlap in 3-D shape/feature space, resulting in relatively larger ST and CT similarity scores. The structure clustering tool is specifically geared towards helping to identify such structure-activity trends in 3-D similarity (as well as 2-D similarity) space and, in combination with the PubChem3D viewer, allow them to be visualized. If one thinks about this, it shows how easy it might be to &#8220;scaffold hop&#8221; or relate diverse chemical structures with similar biological function by examining 3-D similar chemicals in PubChem. It may also suggest that one may be able to better understand additional biological functions of known drugs (<em>i.e.</em>, so called &#8220;side effects&#8221;) by examining their PubChem 3-D similarity to other chemicals with known biological roles.</p>
<div class="figure align-center" id="figure-7">
<img alt="../_images/1758-2946-3-32-7.jpg" src="../_images/1758-2946-3-32-7.jpg" />
<p class="caption">Figure 7. Similarity score matrix for selected histamine H1 receptor antagonist anti-inflammatory drugs</p>
<div class="legend">
<strong>Similarity score matrix for selected histamine H1 receptor antagonist anti-inflammatory drugs</strong>. The lower triangle of the score matrix corresponds to the 2-D similarity computed using the PubChem fingerprint. The upper triangle corresponds to the 3-D similarity ST/CT scores. The matrix elements in red text indicate a 2-D similarity ≥ 0.75 or 3-D similarity with ST ≥ 0.74 and ComboT ≥ 1.0. The first ten diverse conformers per molecule were superimposed using shape-based optimization and the single conformer-pair per compound-pair with the largest ComboT retained.</div>
</div>
<div class="figure align-center" id="figure-8">
<img alt="../_images/1758-2946-3-32-8.jpg" src="../_images/1758-2946-3-32-8.jpg" />
<p class="caption">Figure 8. 3-D superposition of selected histamine H1 receptor antagonist anti-inflammatory drugs</p>
<div class="legend">
<strong>3-D superposition of selected histamine H1 receptor antagonist anti-inflammatory drugs</strong>. Although there is little 2-D similarity, using the PubChem fingerprint, substantial 3-D similarity is found between various structurally diverse anti-inflammatory drugs.</div>
</div>
</div>
</div>
<div class="section" id="id91">
<h2>Conclusions<a class="headerlink" href="#id91" title="Permalink to this headline">¶</a></h2>
<p>A new resource for scientists, PubChem3D, layered on top of PubChem, provides a new dimension to its ability to search, subset, export, visualize, and analyze chemical structures and their associated biological data. With a broad suite of tools and capabilities, 3-D similarity is given equal footing to assist in finding non-obvious trends in experimentally observed biological activity. As a complement to 2-D similarity, 3-D similarity demonstrates a capability to relate chemical series that are not sufficiently 2-D similar.</p>
</div>
<div class="section" id="abbreviations">
<h2>Abbreviations<a class="headerlink" href="#abbreviations" title="Permalink to this headline">¶</a></h2>
<p>2-D: (2-dimensional); 3-D: (3-dimensional); MMFF: (Merck Molecular Force Field); RMSD: (root-mean-square distance).</p>
</div>
<div class="section" id="competing-interests">
<h2>Competing interests<a class="headerlink" href="#competing-interests" title="Permalink to this headline">¶</a></h2>
<p>The authors declare that they have no competing interests.</p>
</div>
<div class="section" id="authors-contributions">
<h2>Authors&#8217; contributions<a class="headerlink" href="#authors-contributions" title="Permalink to this headline">¶</a></h2>
<p>All authors contributed in a material way to the PubChem3D project. Specific attributable contributions are as follows: EEB drafted the manuscript and performed all major project aspects not directly attributed to others; JC implemented the web-based viewer and search interfaces; SK drafted the manuscript and helped develop neighboring accelerators; LH and YS developed analysis database components; WS and SH developed storage and neighboring database components; VS developed the image generation methodology, PC3D viewer application, search backends, and viewer backends; PAT developed the download and score matrix facilities; JW developed the structure clustering and heat-map facilities; BY and PAT developed the identifier exchange facilities; JZ developed the compound/substance summary facilities; and SHB heads the PubChem project. All authors read and approved the final manuscript.</p>
</div>
<div class="section" id="acknowledgements">
<h2>Acknowledgements<a class="headerlink" href="#acknowledgements" title="Permalink to this headline">¶</a></h2>
<p>This research was supported (in part) by the Intramural Research Program of the National Library of Medicine, National Institutes of Health, U. S. Department of Health and Human Services. This effort utilized the high-performance computational capabilities of the Helix Systems at the National Institutes of Health, Bethesda, MD (<a class="reference external" href="http://helix.nih.gov">http://helix.nih.gov</a>).</p>
<p>EEB is very appreciative of the folks at OpenEye Scientific Software, Inc., for allowing their tools and methodologies to be utilized in the PubChem3D project and for the many fruitful suggestions/discussions/insights. EEB is also very thankful for the significant and continuing support of the NCBI systems staff, specifically Ron Patterson, Charlie Cook, and Don Preuss, without which the PubChem3D project would not exist.</p>
<p>We are very grateful to the reviewers for their careful consideration of this manuscript and useful suggestions.</p>
<table class="docutils footnote" frame="void" id="b1" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label">[1]</td><td><em>(<a class="fn-backref" href="#id2">1</a>, <a class="fn-backref" href="#id81">2</a>, <a class="fn-backref" href="#id82">3</a>, <a class="fn-backref" href="#id84">4</a>)</em> PubChem: integrated platform of small molecules and biological activities</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b2" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id3">[2]</a></td><td>PubChem: a public information system for analyzing bioactivities of small molecules</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b3" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id4">[3]</a></td><td>An overview of the PubChem BioAssay resource</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b4" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label">[4]</td><td><em>(<a class="fn-backref" href="#id5">1</a>, <a class="fn-backref" href="#id76">2</a>)</em> Database resources of the National Center for Biotechnology Information</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b5" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label">[5]</td><td><em>(<a class="fn-backref" href="#id6">1</a>, <a class="fn-backref" href="#id27">2</a>)</em> PubChem3D Thematic Series</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b6" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label">[6]</td><td><em>(<a class="fn-backref" href="#id7">1</a>, <a class="fn-backref" href="#id28">2</a>, <a class="fn-backref" href="#id33">3</a>, <a class="fn-backref" href="#id47">4</a>, <a class="fn-backref" href="#id52">5</a>, <a class="fn-backref" href="#id55">6</a>)</em> PubChem3D: conformer generation</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b7" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label">[7]</td><td><em>(<a class="fn-backref" href="#id8">1</a>, <a class="fn-backref" href="#id29">2</a>, <a class="fn-backref" href="#id34">3</a>, <a class="fn-backref" href="#id54">4</a>)</em> PubChem3D: diversity of shape</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b8" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label">[8]</td><td><em>(<a class="fn-backref" href="#id9">1</a>, <a class="fn-backref" href="#id25">2</a>, <a class="fn-backref" href="#id30">3</a>, <a class="fn-backref" href="#id35">4</a>, <a class="fn-backref" href="#id65">5</a>, <a class="fn-backref" href="#id66">6</a>, <a class="fn-backref" href="#id71">7</a>, <a class="fn-backref" href="#id73">8</a>)</em> PubChem3D: similar conformers</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b9" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label">[9]</td><td><em>(<a class="fn-backref" href="#id10">1</a>, <a class="fn-backref" href="#id31">2</a>, <a class="fn-backref" href="#id78">3</a>)</em> PubChem3D: shape compatibility filtering using molecular shape quadrupoles</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b10" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label">[10]</td><td><em>(<a class="fn-backref" href="#id11">1</a>, <a class="fn-backref" href="#id26">2</a>, <a class="fn-backref" href="#id32">3</a>, <a class="fn-backref" href="#id83">4</a>, <a class="fn-backref" href="#id88">5</a>)</em> PubChem3D: biologically relevant 3-D similarity</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b11" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label">[11]</td><td><em>(<a class="fn-backref" href="#id12">1</a>, <a class="fn-backref" href="#id62">2</a>, <a class="fn-backref" href="#id63">3</a>, <a class="fn-backref" href="#id69">4</a>)</em> A Gaussian description of molecular shape</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b12" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label">[12]</td><td><em>(<a class="fn-backref" href="#id13">1</a>, <a class="fn-backref" href="#id79">2</a>)</em> A fast method of molecular shape comparison: a simple application of a Gaussian description of molecular shape</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b13" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id14">[13]</a></td><td>Gaussian shape methods</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b14" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label">[14]</td><td><em>(<a class="fn-backref" href="#id15">1</a>, <a class="fn-backref" href="#id74">2</a>)</em> ROCS - Rapid Overlay of Chemical Structures, Version 2.2</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b15" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label">[15]</td><td><em>(<a class="fn-backref" href="#id16">1</a>, <a class="fn-backref" href="#id57">2</a>, <a class="fn-backref" href="#id58">3</a>, <a class="fn-backref" href="#id60">4</a>, <a class="fn-backref" href="#id61">5</a>, <a class="fn-backref" href="#id75">6</a>)</em> ShapeTK - C++, Version 1.8.0</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b16" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id17">[16]</a></td><td>Synopsis of some recent tactical application of bioisosteres in drug design</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b17" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id18">[17]</a></td><td>Molecular shape and medicinal chemistry: a perspective</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b18" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id19">[18]</a></td><td>Docking: successes and challenges</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b19" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id20">[19]</a></td><td>Structure-based discovery of antibacterial drugs</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b20" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id21">[20]</a></td><td>Structure-based drug design strategies in medicinal chemistry</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b21" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id22">[21]</a></td><td>Structure-based design of molecular cancer therapeutics</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b22" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id23">[22]</a></td><td>Structure-based drug metabolism predictions for drug design</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b23" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id24">[23]</a></td><td>Structure-based strategies for drug design and discovery</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b24" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label">[24]</td><td><em>(<a class="fn-backref" href="#id36">1</a>, <a class="fn-backref" href="#id44">2</a>)</em> Merck molecular force field. 1. Basis, form, scope, parameterization, and performance of MMFF94</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b25" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label">[25]</td><td><em>(<a class="fn-backref" href="#id37">1</a>, <a class="fn-backref" href="#id45">2</a>)</em> Merck molecular force field. 2. MMFF94 van der Waals and electrostatic parameters for intermolecular interactions</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b26" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label">[26]</td><td><em>(<a class="fn-backref" href="#id38">1</a>, <a class="fn-backref" href="#id46">2</a>)</em> MMFF VI. MMFF94s option for energy minimization studies</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b27" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id39">[27]</a></td><td>OMEGA, Version 2.0</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b28" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id40">[28]</a></td><td>OMEGA, Version 2.1</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b29" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id41">[29]</a></td><td>OMEGA, Version 2.2</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b30" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id42">[30]</a></td><td>OMEGA, Version 2.3</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b31" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id43">[31]</a></td><td>OMEGA, Version 2.4</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b32" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id48">[32]</a></td><td>Biasing conformational ensembles towards bioactive-like conformers for ligand-based drug design</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b33" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id49">[33]</a></td><td>Conformer generation with OMEGA: algorithm and validation using high quality structures from the Protein Databank and Cambridge Structural Database</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b34" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id50">[34]</a></td><td>MIMUMBA revisited: Torsion angle rules for conformer generation derived from X-ray structures</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b35" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id51">[35]</a></td><td>Reproducing the conformations of protein-bound ligands: a critical evaluation of several popular conformational searching tools</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b36" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label">[36]</td><td><em>(<a class="fn-backref" href="#id53">1</a>, <a class="fn-backref" href="#id56">2</a>, <a class="fn-backref" href="#id70">3</a>)</em> OEChem, Version 1.7.4</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b37" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id59">[37]</a></td><td>Three-dimensional hydrogen-bond geometry and probability information from a crystal survey</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b38" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id64">[38]</a></td><td>Szybki TK, Version 1.5.0</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b39" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id67">[39]</a></td><td>Clustering of chemical structures on the basis of two-dimensional similarity measures</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b40" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id68">[40]</a></td><td>Implementing drug screening programs using molecular similarity methods</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b41" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id72">[41]</a></td><td>Fast 3D shape screening of large chemical databases through alignment-recycling</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b42" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id80">[42]</a></td><td>A new class of molecular shape descriptors. 1. Theory and properties</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b43" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id85">[43]</a></td><td>Entrez Programming Utilities Help</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b44" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id86">[44]</a></td><td>Taverna: a tool for building and running workflows of services</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b45" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id87">[45]</a></td><td>Pipeline Pilot, Version 8.5</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b46" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id89">[46]</a></td><td>HTS identification of compounds activating TNAP at an intermediate concentration of phosphate acceptor detected in luminescent assay</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b47" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id90">[47]</a></td><td>Medical Subject Headings</td></tr>
</tbody>
</table>
</div>
</div>


          </div>
        </div>
      </div>
        </div>
        <div class="sidebar">
          <h3>Table Of Contents</h3>
          <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-1.html">CurlySMILES: a chemical language to customize and annotate encodings of molecular and nanodevice structures</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-2.html">Prediction of cyclin-dependent kinase 2 inhibitor potency using the fragment molecular orbital method</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-3.html">jCompoundMapper: An open source Java library and command-line tool for chemical fingerprints</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-4.html">PubChem3D: Conformer generation</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-5.html">Modular Chemical Descriptor Language (MCDL): Stereochemical modules</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-6.html">FlaME: Flash Molecular Editor - a 2D structure input tool for the web</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-7.html">Use of structure-activity landscape index curves and curve integrals to evaluate the performance of multiple machine learning prediction models</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-8.html">Confab - Systematic generation of diverse low-energy conformers</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-9.html">PubChem3D: Diversity of shape</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-11.html">Interpreting linear support vector machine models with heat map molecule coloring</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-12.html">Multilevel Parallelization of AutoDock 4.2</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-13.html">PubChem3D: Similar conformers</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-14.html">Analysis of <em>in vitro</em> bioactivity data extracted from drug discovery literature and patents: Ranking 1654 human protein targets by assayed compounds and molecular scaffolds</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-15.html">Resource description framework technologies in chemistry</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-16.html">Semantic Web integration of Cheminformatics resources with the SADI framework</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-17.html">ChemicalTagger: A tool for semantic text-mining in chemistry</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-18.html">AMBIT RESTful web services: an implementation of the OpenTox application programming interface</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-19.html">Linked open drug data for pharmaceutical research and development</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-20.html">Chemical Entity Semantic Specification: Knowledge representation for efficient semantic cheminformatics and facile data integration</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-21.html">Consistent two-dimensional visualization of protein-ligand complex series</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-22.html">Predicting a small molecule-kinase interaction map: A machine learning approach</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-23.html">4D Flexible Atom-Pairs: An efficient probabilistic conformational space comparison for ligand-based virtual screening</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-24.html">Data governance in predictive toxicology: A review</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-25.html">PubChem3D: Shape compatibility filtering using molecular shape quadrupoles</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-26.html">PubChem3D: Biologically relevant 3-D similarity</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-27.html">Theoretical NMR correlations based Structure Discussion</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-28.html">AZOrange - High performance open source machine learning for QSAR modeling in a graphical programming environment</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-29.html">Multiple search methods for similarity-based virtual screening: analysis of search overlap and precision</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-30.html">Structural diversity of biologically interesting datasets: a scaffold analysis approach</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-31.html">Statistical filtering for NMR based structure generation</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="">PubChem3D: a new resource for scientists</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#abstract">Abstract</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#background">Background</a></li>
<li class="toctree-l3"><a class="reference internal" href="#description">Description</a></li>
<li class="toctree-l3"><a class="reference internal" href="#conclusions">Conclusions</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#id1">Background</a></li>
<li class="toctree-l2"><a class="reference internal" href="#construction-and-content">Construction and Content</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#pubchem3d-coverage">1. PubChem3D Coverage</a></li>
<li class="toctree-l3"><a class="reference internal" href="#conformer-models">2. Conformer Models</a></li>
<li class="toctree-l3"><a class="reference internal" href="#conformer-model-properties">3. Conformer Model Properties</a></li>
<li class="toctree-l3"><a class="reference internal" href="#similar-conformer-neighboring-relationship">4. Similar Conformer Neighboring Relationship</a></li>
<li class="toctree-l3"><a class="reference internal" href="#ftp-site">5. FTP Site</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#utility">Utility</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#ncbi-entrez-interface">1. NCBI Entrez Interface</a></li>
<li class="toctree-l3"><a class="reference internal" href="#visualization">2. Visualization</a></li>
<li class="toctree-l3"><a class="reference internal" href="#search">3. Search</a></li>
<li class="toctree-l3"><a class="reference internal" href="#download">4. Download</a></li>
<li class="toctree-l3"><a class="reference internal" href="#similarity-computation">5. Similarity Computation</a></li>
<li class="toctree-l3"><a class="reference internal" href="#clustering-and-analysis">6. Clustering and Analysis</a></li>
<li class="toctree-l3"><a class="reference internal" href="#programmatic-interface">7. Programmatic Interface</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#examples-of-use">Examples of use</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#finding-additional-biological-annotation">1. Finding additional biological annotation</a></li>
<li class="toctree-l3"><a class="reference internal" href="#relating-chemical-probes-for-same-biological-target">2. Relating chemical probes for same biological target</a></li>
<li class="toctree-l3"><a class="reference internal" href="#relating-chemically-diverse-structures-with-same-pharmacological-action">3. Relating chemically diverse structures with same pharmacological action</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#id91">Conclusions</a></li>
<li class="toctree-l2"><a class="reference internal" href="#abbreviations">Abbreviations</a></li>
<li class="toctree-l2"><a class="reference internal" href="#competing-interests">Competing interests</a></li>
<li class="toctree-l2"><a class="reference internal" href="#authors-contributions">Authors&#8217; contributions</a></li>
<li class="toctree-l2"><a class="reference internal" href="#acknowledgements">Acknowledgements</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-33.html">Open Babel: An open chemical toolbox</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-34.html">Adventures in public data</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-35.html">Three stories about the conduct of science: Past, future, and present</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-36.html">Openness as infrastructure</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-37.html">Open Data, Open Source and Open Standards in chemistry: The Blue Obelisk five years on</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-38.html">The Quixote project: Collaborative and Open Quantum Chemistry data management in the Internet age</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-39.html">CMLLite: a design philosophy for CML</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-40.html">Mining chemical information from open patents</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-41.html">OSCAR4: a flexible architecture for chemical text-mining</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-42.html">The semantic architecture of the World-Wide Molecular Matrix (WWMM)</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-43.html">The semantics of Chemical Markup Language (CML): dictionaries and conventions</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-44.html">CML: Evolution and design</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-45.html">Ami - The chemist&#8217;s amanuensis</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-46.html">The past, present and future of Scientific discourse</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-47.html">Open Bibliography for Science, Technology, and Medicine</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-48.html">Semantic science and its communication - a personal view</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-49.html">Molecular dynamics simulations and in silico peptide ligand screening of the Elk-1 ETS domain</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-50.html">2D-Qsar for 450 types of amino acid induction peptides with a novel substructure pair descriptor having wider scope</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-51.html">An investigation into pharmaceutically relevant mutagenicity data and the influence on Ames predictive potential</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-52.html">Automated annotation of chemical names in the literature with tunable accuracy</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-53.html">MyChemise: A 2D drawing program that uses morphing for visualisation purposes</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-54.html">New developments on the cheminformatics open workflow environment CDK-Taverna</a></li>
</ul>

          <h3 style="margin-top: 1.5em;">Search</h3>
          <form class="search" action="../search.html" method="get">
            <input type="text" name="q" />
            <input type="submit" value="Go" />
            <input type="hidden" name="check_keywords" value="yes" />
            <input type="hidden" name="area" value="default" />
          </form>
          <p class="searchtip" style="font-size: 90%">
            Enter search terms or a module, class or function name.
          </p>
        </div>
        <div class="clearer"></div>
      </div>
    </div>

    <div class="footer-wrapper">
      <div class="footer">
        <div class="left">
          <a href="1758-2946-3-31.html" title="Statistical filtering for NMR based structure generation"
             >previous</a> |
          <a href="1758-2946-3-33.html" title="Open Babel: An open chemical toolbox"
             >next</a> |
          <a href="../genindex.html" title="General Index"
             >index</a>
            <br/>
            <a href="../_sources/articles/1758-2946-3-32.txt"
               rel="nofollow">Show Source</a>
        </div>

        <div class="right">
          
    <div class="footer">
        &copy; Copyright 2012, Noel O&#39;Boyle.
      Created using <a href="http://sphinx.pocoo.org/">Sphinx</a> 1.1.3.
    </div>
        </div>
        <div class="clearer"></div>
      </div>
    </div>

  </body>
</html>