

<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>New developments on the cheminformatics open workflow environment CDK-Taverna &mdash; Papers 1.0 documentation</title>
    
    <link rel="stylesheet" href="../_static/agogo.css" type="text/css" />
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '../',
        VERSION:     '1.0',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="../_static/jquery.js"></script>
    <script type="text/javascript" src="../_static/underscore.js"></script>
    <script type="text/javascript" src="../_static/doctools.js"></script>
    <link rel="top" title="Papers 1.0 documentation" href="../index.html" />
    <link rel="prev" title="MyChemise: A 2D drawing program that uses morphing for visualisation purposes" href="1758-2946-3-53.html" /> 
  </head>
  <body>
    <div class="header-wrapper">
      <div class="header">
        <div class="headertitle"><a
          href="../index.html">Papers 1.0 documentation</a></div>
        <div class="rel">
          <a href="1758-2946-3-53.html" title="MyChemise: A 2D drawing program that uses morphing for visualisation purposes"
             accesskey="P">previous</a> |
          <a href="../genindex.html" title="General Index"
             accesskey="I">index</a>
        </div>
       </div>
    </div>

    <div class="content-wrapper">
      <div class="content">
        <div class="document">
            
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body">
            
  <div class="section" id="new-developments-on-the-cheminformatics-open-workflow-environment-cdk-taverna">
<h1>New developments on the cheminformatics open workflow environment CDK-Taverna<a class="headerlink" href="#new-developments-on-the-cheminformatics-open-workflow-environment-cdk-taverna" title="Permalink to this headline">¶</a></h1>
<div class="section" id="abstract">
<h2>Abstract<a class="headerlink" href="#abstract" title="Permalink to this headline">¶</a></h2>
<div class="section" id="background">
<h3>Background<a class="headerlink" href="#background" title="Permalink to this headline">¶</a></h3>
<p>The computational processing and analysis of small molecules is at heart of cheminformatics and structural bioinformatics and their application in e.g. metabolomics or drug discovery. Pipelining or workflow tools allow for the Lego™-like, graphical assembly of I/O modules and algorithms into a complex workflow which can be easily deployed, modified and tested without the hassle of implementing it into a monolithic application. The CDK-Taverna project aims at building a free open-source cheminformatics pipelining solution through combination of different open-source projects such as Taverna, the Chemistry Development Kit (CDK) or the Waikato Environment for Knowledge Analysis (WEKA). A first integrated version 1.0 of CDK-Taverna was recently released to the public.</p>
</div>
<div class="section" id="results">
<h3>Results<a class="headerlink" href="#results" title="Permalink to this headline">¶</a></h3>
<p>The CDK-Taverna project was migrated to the most up-to-date versions of its foundational software libraries with a complete re-engineering of its worker&#8217;s architecture (version 2.0). 64-bit computing and multi-core usage by paralleled threads are now supported to allow for fast in-memory processing and analysis of large sets of molecules. Earlier deficiencies like workarounds for iterative data reading are removed. The combinatorial chemistry related reaction enumeration features are considerably enhanced. Additional functionality for calculating a natural product likeness score for small molecules is implemented to identify possible drug candidates. Finally the data analysis capabilities are extended with new workers that provide access to the open-source WEKA library for clustering and machine learning as well as training and test set partitioning. The new features are outlined with usage scenarios.</p>
</div>
<div class="section" id="conclusions">
<h3>Conclusions<a class="headerlink" href="#conclusions" title="Permalink to this headline">¶</a></h3>
<p>CDK-Taverna 2.0 as an open-source cheminformatics workflow solution matured to become a freely available and increasingly powerful tool for the biosciences. The combination of the new CDK-Taverna worker family with the already available workflows developed by a lively Taverna community and published on myexperiment.org enables molecular scientists to quickly calculate, process and analyse molecular data as typically found in e.g. today&#8217;s systems biology scenarios.</p>
</div>
</div>
<div class="section" id="id1">
<h2>Background<a class="headerlink" href="#id1" title="Permalink to this headline">¶</a></h2>
<p>Current problems in the biosciences typically involve several domains of research. They require a scientist to work with different and diverse sets of data. The reconstruction of a metabolic network from sequencing data, for example, employs many of the data types found along the axis of the central dogma, including reconstruction of genome sequences, gene prediction, determination of encoded protein families, and from there to the substrates of enzymes, which then form the metabolic network. In order to work with such a processing pipeline, a scientist has to copy/paste and often transform the data between several bioinformatics web portals by hand. The manual approach involves repetitive tasks and cannot be considered effective or scalable.</p>
<p>Especially the processing and analysis of small molecules comprises tasks like filtering, transformation, curation or migration of chemical data, information retrieval with substructures, reactions, or pharmacophores as well as the analysis of molecular data with statistics, clustering or machine learning to support chemical diversity requirements or to generate quantitative structure activity/property relationships (QSAR/QSPR models). These processing and analysis procedures itself are of increasing importance for research areas like metabolomics or drug discovery. The power and flexibility of the corresponding computational tools become essential success factors for the whole research process.</p>
<p>The workflow paradigm addresses the above issues with the supply of sets of elementary workers (activities) that can be flexibly assembled in a graphical manner to allow complex procedures to be performed in an effective manner - without the need of specific code development or software programming skills. Scientific workflows allow the combination of a wide spectrum of algorithms and resources in a single workspace <a class="footnote-reference" href="#b1" id="id2">[1]</a><a class="footnote-reference" href="#b2" id="id3">[2]</a><a class="footnote-reference" href="#b3" id="id4">[3]</a>. Earlier problems with iterations over large data sets <a class="footnote-reference" href="#b4" id="id5">[4]</a> are completely resolved in version 2.0 due to new implementations in Taverna. Taverna 2 allows control structures such as &#8220;while&#8221; loops or &#8220;if-then-else&#8221; constructs. Termination criteria for loops may now be evaluated by listening to a state port <a class="footnote-reference" href="#b5" id="id6">[5]</a>. In addition the user interface of the Taverna 2 workbench has clearly improved: The design and manipulation of workflows in a graphical workflow editor is now supported. Features like copy/paste and undo/redo simplify workflow creation and maintenance <a class="footnote-reference" href="#b6" id="id7">[6]</a>.</p>
<p>The CDK-Taverna project aims at building a free open-source cheminformatics pipelining solution through combination of different open-source projects such as Taverna <a class="footnote-reference" href="#b7" id="id8">[7]</a>, the Chemistry Development Kit (CDK) <a class="footnote-reference" href="#b8" id="id9">[8]</a><a class="footnote-reference" href="#b9" id="id10">[9]</a>, or the Waikato Environment for Knowledge Analysis (WEKA) <a class="footnote-reference" href="#b10" id="id11">[10]</a>. A first integrated version 1.0 of CDK-Taverna was recently released to the public <a class="footnote-reference" href="#b4" id="id12">[4]</a>. To extend usability and power of CDK-Taverna for different molecular research purposes the development of version 2.0 was motivated.</p>
<div class="section" id="implementation">
<h3>Implementation<a class="headerlink" href="#implementation" title="Permalink to this headline">¶</a></h3>
<p>The CDK-Taverna 2.0 plug-in makes use of the Taverna plug-in manager for its installation. The manager fetches all necessary information about the plug-in from a XML file which is located at <a class="reference external" href="http://www.ts-concepts.de/cdk-taverna2/plugin/">http://www.ts-concepts.de/cdk-taverna2/plugin/</a>. The information provided therein contains the name of the plug-in, its version, the repository location and the required Taverna version. Upon submitting the URL to the plug-in manager it downloads all necessary dependencies automatically from the web. After a subsequent restart the plug-in is enabled and the workers are visible in the services. The plug-in uses Taverna version 2.2.1 <a class="footnote-reference" href="#b6" id="id13">[6]</a>, CDK version 1.3.8 <a class="footnote-reference" href="#b11" id="id14">[11]</a> and WEKA version 3.6.4 <a class="footnote-reference" href="#b12" id="id15">[12]</a>. Like its predecessor it uses the Maven 2 build system <a class="footnote-reference" href="#b13" id="id16">[13]</a> as well as the Taverna workbench for automated dependency management.</p>
</div>
<div class="section" id="cdk-taverna-2-0-worker-implementation">
<h3>CDK-Taverna 2.0 worker implementation<a class="headerlink" href="#cdk-taverna-2-0-worker-implementation" title="Permalink to this headline">¶</a></h3>
<p>The CDK-Taverna 2.0 plug-in is designed to be easily extendible: The implementation allows to create new workers by simply inheriting from the single abstract class</p>
<p>•</p>
<p>•</p>
<p>•</p>
<p>•</p>
<p>•</p>
<p>Finally a new worker has to be registered to be available in the Taverna workbench. For this purpose Taverna offers the class</p>
<p>Besides the basic implementation it is possible to define a configuration panel for a worker which allows the specification of parameters. A configuration panel has to inherit from the abstract class</p>
<p>•</p>
<p>•</p>
<p>•</p>
<p>•</p>
<p>•</p>
<p>The configuration panel has to be registered in the <a class="reference external" href="http://cdk-taverna-2.ts-concepts.de/wiki/index.php?title=Main_Page">http://cdk-taverna-2.ts-concepts.de/wiki/index.php?title=Main_Page</a>.</p>
</div>
<div class="section" id="requirements">
<h3>Requirements<a class="headerlink" href="#requirements" title="Permalink to this headline">¶</a></h3>
<p>CDK-Taverna 2.0 supports 64-bit computing by use with a Java 64-bit virtual machine. The CDK-Taverna 2.0 plug-in is written in Java and requires Java 6 or higher. The latest Java version is available at <a class="reference external" href="http://www.java.com/de/download/">http://www.java.com/de/download/</a>. The CDK-Taverna 2.0 plug-in is developed and tested on Microsoft Windows 7 as well as Linux and Mac OS/X (32 and 64-bit).</p>
</div>
</div>
<div class="section" id="results-and-discussion">
<h2>Results and Discussion<a class="headerlink" href="#results-and-discussion" title="Permalink to this headline">¶</a></h2>
<p>The CDK-Taverna 2.0 plug-in provides 192 workers for input and output (I/O) of various chemical file and line notation formats, substructure filtering, aromaticity detection, atom typing, reaction enumeration, molecular descriptor calculation and data analysis. Parallel computing with multi-core processors by use of multiple concurrent threads is flexibly implemented for many workers where operations scale nearly linear with the number of cores. Especially the machine learning and the molecular descriptor calculation workers benefit from parallel computation. An overview is given in Tables <em class="xref std std-ref">1</em> and <em class="xref std std-ref">2</em>. Many workers are described by example workflows available at <a class="reference external" href="http://cdk-taverna-2.ts-concepts.de/wiki/index.php?title=Main_Page">http://cdk-taverna-2.ts-concepts.de/wiki/index.php?title=Main_Page</a>. Additionally, the workflows can be found at <a class="reference external" href="http://www.myexperiment.org/">http://www.myexperiment.org/</a>.</p>
<p>CDK-Taverna 1.0 was confined to 32-bit Java virtual machine and thus was restricted to in-memory processing of data volumes of at most 2 gigabyte in practice. Version 2.0 also supports 64-bit computing by use of a 64-bit Java virtual machine so that the processable data volume is only limited by hardware constraints (memory, speed): 64-bit in-memory workflows were successfully performed with data sets of about 1 million small molecules. Since the memory restrictions of version 1.0 were a main reason to use Pgchem::tigress as a molecular database backend <a class="footnote-reference" href="#b4" id="id18">[4]</a> the corresponding version 1.0 workers were not migrated to the current version 2.0 yet.</p>
<div class="section" id="advanced-reaction-enumeration">
<h3>Advanced reaction enumeration<a class="headerlink" href="#advanced-reaction-enumeration" title="Permalink to this headline">¶</a></h3>
<p>CDK-Taverna 1.0 provided basic functions for combinatorial chemistry related reaction enumeration: They supported the use of two reactants, a single product and one generic group per reactant. The new enumeration options used by CDK-Taverna 2.0 offer major enhancements like multi-match detection, any number of reactants, products or generic groups as well as variable R-groups, ring sizes and atom definitions. The extended functionality was developed and applied in industrial cooperation projects. Advanced reaction enumeration features are illustrated in Figure <a class="reference internal" href="1758-2946-3-46.html#figure-1"><em>1</em></a>. The <em>Variable RGroup</em> feature allows the definition of chemical groups which can be flexibly attached to predefined atoms with syntax [A:B,B,B...-RC] where <em>A</em> is a freely selectable identifier, <em>B</em> are numbers from an <em>Atom-to-Atom-Mapping</em> defining the atoms to which the generic group can be attached and <em>C</em> is the chemical group identifier which can be any number. The <em>Atom Alias</em> feature offers the possibility to define a wild card for preconfigured elements. The syntax is [A:B,B,B...] where <em>A</em> is a freely selectable identifier and <em>B</em> are the string representations of the possible elements. The <em>Expandable Atom</em> feature enables the definition of freely sizeable rings or aliphatic chains with syntax [A:[]B] where <em>A</em> is a freely selectable identifier and <em>B</em> is the maximum number of atoms to insert. Figure <a class="reference internal" href="1758-2946-3-46.html#figure-2"><em>2</em></a> depicts a workflow for reaction enumeration. The capabilities of the advanced reaction enumerator implementation are summarized in Figure <a class="reference internal" href="1758-2946-3-46.html#figure-3"><em>3</em></a> which also demonstrates multi-match detection, i.e. multiple reaction centers within one molecule.</p>
<div class="figure align-center" id="figure-1">
<img alt="../_images/1758-2946-3-54-1.jpg" src="../_images/1758-2946-3-54-1.jpg" />
<p class="caption">Figure 1. Advanced reaction enumeration features: (left) The <em>Variable RGroup</em> feature allows the definition of chemical groups which can be flexibly attached to predefined atoms</p>
<div class="legend">
<strong>Variable RGroup Advanced reaction enumeration features: (left) The</strong> . (middle) The Atom Alias feature offers the possibility to define a wild card for preconfigured elements. (right) The Expandable Atom feature enables the definition of freely sizeable rings or aliphatic chains.</div>
</div>
<div class="figure align-center" id="figure-2">
<img alt="../_images/1758-2946-3-54-2.jpg" src="../_images/1758-2946-3-54-2.jpg" />
<p class="caption">Figure 2. Workflow for reaction enumeration: After loading a generic reaction (</p>
<div class="legend">
<strong>Workflow for reaction enumeration: After loading a generic reaction (**</strong>, from a MDL RXN file) and two educt lists (<a href="#id19"><span class="problematic" id="id20">**</span></a><strong>, from MDL SD files) the</strong> <strong>worker performs the enumeration with the results stored as MDL RXN files</strong>. An additional PDF file is created which shows all enumerated reactions in a tabular manner. The results are stored in the output folder determined by the OUT input port.</div>
</div>
<div class="figure align-center" id="figure-3">
<img alt="../_images/1758-2946-3-54-3.jpg" src="../_images/1758-2946-3-54-3.jpg" />
<p class="caption">Figure 3. Capabilities of the advanced reaction enumerator: The sketched generic reaction contains three different generic groups labelled X, Y and Z</p>
<div class="legend">
<strong>Capabilities of the advanced reaction enumerator: The sketched generic reaction contains three different generic groups labelled X, Y and Z</strong>. Group × defines a <em>Variable RGroup</em> which can freely attach to all atoms of the ring. The <em>Atom Alias</em> group labelled Y is a wild card for the elements carbon, oxygen and nitrogen. The <em>Expandable Atom</em> group Z defines a variable ring size: The ring can be expanded by up to two additional carbon atoms. The enumerated products with the small letters <em>a</em> and <em>b</em> originate from <em>multi-match detection</em>.</div>
</div>
</div>
<div class="section" id="evaluation-of-small-molecules-for-natural-product-likeness">
<h3>Evaluation of small molecules for natural product likeness<a class="headerlink" href="#evaluation-of-small-molecules-for-natural-product-likeness" title="Permalink to this headline">¶</a></h3>
<p>In recent years, computer assisted drug design studies use natural product (NP) likeness as a criterion to screen compound libraries for potential drug candidates <a class="footnote-reference" href="#b14" id="id21">[14]</a><a class="footnote-reference" href="#b15" id="id22">[15]</a>. The reason to estimate NP likeness during candidate screening is to facilitate the selection of those compounds that mimic structural features that are naturally evolved to best interact with biological targets.</p>
<p>Version 2.0 of CDK-Taverna provides two groups of workers that re-implement the work of <em>Ertl et al</em> to score small molecules for NP-likeness <a class="footnote-reference" href="#b14" id="id23">[14]</a>. The workers in the Molecule Curation folder are dedicated to the pre-processing of chemical structures: The Molecule Connectivity Checker worker removes counter ions and disconnects fragments, the Remove Sugar Groups worker removes all sugar rings and linear sugars from structures and the Curate Strange Elements worker discards structures that are composed of elements other than non-metals. This set of curation workers finally creates scaffolds olds and sub structures. From these structures atom signatures <a class="footnote-reference" href="#b16" id="id24">[16]</a> are generated using the Generate Atom Signatures worker and exploited as structural descriptors in charting the compound&#8217;s region in the chemical structure space. The combined workflow of curation and atom signature generation workers is illustrated in Figure <a class="reference internal" href="1758-2946-3-46.html#figure-4"><em>4</em></a>. Using this workflow, atom signatures can be generated for user-defined training (Natural products and synthetics) and testing (compound libraries) structural dataset. Workers of the Signature Scoring folder use atom signatures generated from compound libraries and rank them for NP-likeness based on the statistics suggested by <em>Ertl et al</em> <a class="footnote-reference" href="#b14" id="id25">[14]</a>. This scoring workflow is illustrated in Figure <a class="reference internal" href="1758-2946-3-46.html#figure-5"><em>5</em></a>. The whole package of workflows is available for free download at <a class="reference external" href="http://www.myexperiment.org/users/10069/packs">http://www.myexperiment.org/users/10069/packs</a>. The curation and signature scoring workers may not only be applied in evaluating the NP-likeness of compound libraries but also in evaluating the metabolite-likeness of theoretical metabolites for predicting whole metabolomes. The latter application was the original purpose for the worker development and corresponding results will be presented in a subsequent publication.</p>
<div class="figure align-center" id="figure-4">
<img alt="../_images/1758-2946-3-54-4.jpg" src="../_images/1758-2946-3-54-4.jpg" />
<p class="caption">Figure 4. Molecule curation and atom signature descriptor generation workflow: The</p>
<div class="legend">
<strong>Molecule curation and atom signature descriptor generation workflow: The</strong> <strong>takes the Structure-Data File (SDF) of compounds (**</strong>) as input and pass the structures down the workflow for molecule curation and atom signature generation**. The number of structures to be read, and pumped down the workflow can be configured (<a class="reference external" href="http://www.myexperiment.org/workflows/2120.html">http://www.myexperiment.org/workflows/2120.html</a>.</div>
</div>
<div class="figure align-center" id="figure-5">
<img alt="../_images/1758-2946-3-54-5.jpg" src="../_images/1758-2946-3-54-5.jpg" />
<p class="caption">Figure 5. NP-likeness scoring workflow: This workflow take inputs of atom signatures file generated from the user defined natural products library (</p>
<div class="legend">
<strong>NP-likeness scoring workflow: This workflow take inputs of atom signatures file generated from the user defined natural products library (**</strong>) as well as synthetics (<a href="#id26"><span class="problematic" id="id27">**</span></a><strong>) and compound libraries (**</strong>) and score the compound libraries (<a href="#id28"><span class="problematic" id="id29">**</span></a><strong>) for NP-likeness</strong>. The higher the score the more is the NP-likeness of a molecule. The <a class="reference external" href="http://www.myexperiment.org/workflows/2121.html">http://www.myexperiment.org/workflows/2121.html</a>.</div>
</div>
</div>
<div class="section" id="clustering-and-machine-learning-applications">
<h3>Clustering and machine learning applications<a class="headerlink" href="#clustering-and-machine-learning-applications" title="Permalink to this headline">¶</a></h3>
<p>Unsupervised clustering tries to partition input data into a number of groups smaller than the number of data whereas supervised machine learning tries to construct model functions that map the input data onto their corresponding output data. If the output codes continuous quantities a regression task is defined. Alternatively the output may code classes so that a classification task is addressed. Molecular data sets for clustering consist of input vectors where each vector represents a molecular entity and consists of a set of molecular descriptors itself. Molecular data sets for machine learning add to each input vector a corresponding output vector with features to be learned - thus they consist of I/O pairs of input and output vectors.</p>
<p>The clustering and machine learning workers of CDK-Taverna 2.0 allow the use of distinct WEKA functionality. As far as clustering is concerned the ART-2a worker of version 1.0 is supplemented with five additional WEKA-based workers which offer</p>
<p>• <a class="footnote-reference" href="#b17" id="id30">[17]</a>.</p>
<p>• <a class="footnote-reference" href="#b18" id="id31">[18]</a>.</p>
<p>• <a class="footnote-reference" href="#b19" id="id32">[19]</a>.</p>
<p>• <a class="footnote-reference" href="#b20" id="id33">[20]</a>.</p>
<p>• <a class="footnote-reference" href="#b21" id="id34">[21]</a>.</p>
<p>Machine learning workers support the significance analysis of single components (i.e. features) of an input vector to obtain smaller inputs with a reduced set of components/features, the partitioning of machine learning data into training and test sets, the construction of input/output mapping model functions and model based predictions as well as result visualization. There is a total of six WEKA-based machine learning methods available: Two workers allow regression as well as classification procedures...</p>
<p>• <a class="footnote-reference" href="#b22" id="id35">[22]</a>.</p>
<p>• <a class="footnote-reference" href="#b23" id="id36">[23]</a>.</p>
<p>... two workers do only support regression...</p>
<p>•</p>
<p>• <a class="footnote-reference" href="#b24" id="id37">[24]</a><a class="footnote-reference" href="#b25" id="id38">[25]</a>.</p>
<p>... and two workers are restricted to classification tasks:</p>
<p>• <a class="footnote-reference" href="#b26" id="id39">[26]</a>.</p>
<p>• <a class="footnote-reference" href="#b27" id="id40">[27]</a>.</p>
<p>For selection of an optimum reduced set of input vector components there are two workers available. The <a class="reference internal" href="1758-2946-3-45.html#figure-6"><em>6</em></a> illustrates the procedure. The <a class="footnote-reference" href="#b28" id="id41">[28]</a>. In each iteration the single component is discarded that has the smallest in influence on the RMSE - up to a last &#8220;most significant&#8221; component. Figure <a class="reference internal" href="1758-2946-3-45.html#figure-7"><em>7</em></a> shows a result of a &#8220;leave-one-out&#8221; analysis and Figure <a class="reference internal" href="1758-2946-3-45.html#figure-8"><em>8</em></a> depicts the related workflow.</p>
<div class="figure align-center" id="figure-6">
<img alt="../_images/1758-2946-3-54-6.jpg" src="../_images/1758-2946-3-54-6.jpg" />
<p class="caption">Figure 6. Genetic algorithm for selection of an optimum reduced set of input vector components: The algorithm starts with a random population in which each chromosome consists of a random distribution of enabled/disabled (on/off) input vector components denoted <em>A</em><sub>1</sub> to <sub>n</sub> A(where the number of components with &#8220;on&#8221; status remains fixed during evolution)</p>
<div class="legend">
<strong>A:sub:`1` to :sub:`n` AGenetic algorithm for selection of an optimum reduced set of input vector components: The algorithm starts with a random population in which each chromosome consists of a random distribution of enabled/disabled (on/off) input vector components denoted</strong> . This distribution is changed by mutation and cross-over. The fitness of each chromosome is evaluated by the inverse square RMSE. The selection process for each generation is performed by Roulette wheel selection where chromosomes are inherited with probabilities that correspond to their particular fitness.</div>
</div>
<div class="figure align-center" id="figure-7">
<img alt="../_images/1758-2946-3-54-7.jpg" src="../_images/1758-2946-3-54-7.jpg" />
<p class="caption">Figure 7. &#8220;Leave-One-Out&#8221; analysis to estimate the significance of input vector components: The root mean square error (RMSE) rises with an increasing number of discarded components (i.e. a decreasing number of input vector components used for the machine filearning procedure)</p>
<div class="legend">
<strong>&#8220;Leave-One-Out&#8221; analysis to estimate the significance of input vector components: The root mean square error (RMSE) rises with an increasing number of discarded components (i.e. a decreasing number of input vector components used for the machine filearning procedure)</strong>. The relative RMSE shift from step to step may be correlated with the significance of the discarded component. In this case it is shown that the first fifty components do only have a negligible in influence on the machine learning result and thus may be excluded from further analysis.</div>
</div>
<div class="figure align-center" id="figure-8">
<img alt="../_images/1758-2946-3-54-8.jpg" src="../_images/1758-2946-3-54-8.jpg" />
<p class="caption">Figure 8. Workflow for &#8220;Leave-One-Out&#8221; analysis: First a regression dataset is generated from a CSV file with UUID and molecular descriptor input data for each molecule (</p>
<div class="legend">
<strong>Workflow for &#8220;Leave-One-Out&#8221; analysis: First a regression dataset is generated from a CSV file with UUID and molecular descriptor input data for each molecule (**</strong>) and a CSV file containing the UUID of the molecule and the corresponding output (regression) value (<a href="#id42"><span class="problematic" id="id43">**</span></a><strong>)</strong>. Then the</div>
</div>
<p>For training and test set partitioning the <a class="footnote-reference" href="#b28" id="id44">[28]</a>:</p>
<p>•</p>
<p>•</p>
<p>•</p>
<p>Figure <a class="reference internal" href="1758-2946-3-45.html#figure-9"><em>9</em></a> shows a workflow using the <a class="reference internal" href="1758-2946-3-45.html#figure-10"><em>10</em></a>. Classification workers may be used in an equivalent manner. Figure <a class="reference internal" href="1758-2946-3-45.html#figure-11"><em>11</em></a> depicts diagrams and output of a QSPR analysis to predict HPLC retention times for small molecules: The experimental dataset consists of 183 I/O pairs with a set of molecular descriptors for each small molecule as an input and the corresponding retention time as an output. The molecular descriptors were calculated with the</p>
<div class="figure align-center" id="figure-9">
<img alt="../_images/1758-2946-3-54-9.jpg" src="../_images/1758-2946-3-54-9.jpg" />
<p class="caption">Figure 9. Partitioning into training and test set: A regression dataset is split into a training and a test set which is performed by the</p>
<div class="legend">
<strong>Partitioning into training and test set: A regression dataset is split into a training and a test set which is performed by the</strong></div>
</div>
<div class="figure align-center" id="figure-10">
<img alt="../_images/1758-2946-3-54-10.jpg" src="../_images/1758-2946-3-54-10.jpg" />
<p class="caption">Figure 10. Configuration panel for the</p>
<div class="legend">
<strong>Configuration panel for the Weka Regression worker: The configuration for a three-layer perceptron neural networks is selected</strong>. Each machine learning method consists of a parameter panel for individual configuration.</div>
</div>
<div class="figure align-center" id="figure-11">
<img alt="../_images/1758-2946-3-54-11.jpg" src="../_images/1758-2946-3-54-11.jpg" />
<p class="caption">Figure 11. Diagrams for machine learning results</p>
<div class="legend">
<strong>Diagrams for machine learning results:</strong> (upper left) Scatter plot with experimental versus predicted output values. (upper right) Residuals plot with differences between the predicted and experimental output values. (lower left) Experimental output data are plotted over corresponding sorted predicted output data. (lower right) Characteristic quantities of the predicted model.</div>
</div>
</div>
<div class="section" id="cdk-taverna-2-0-wiki">
<h3>CDK-Taverna 2.0 Wiki<a class="headerlink" href="#cdk-taverna-2-0-wiki" title="Permalink to this headline">¶</a></h3>
<p>Based on the free MediaWiki framework a Wiki was developed for the CDK-Taverna 2.0 project <a class="footnote-reference" href="#b29" id="id45">[29]</a>. The web page provides general information about the project, documentation about available workers/workflows and on how to create them as well as about installation procedures. The Wiki can be found at <a class="reference external" href="http://cdk-taverna-2.ts-concepts.de/wiki/index.php?title=Main_Page">http://cdk-taverna-2.ts-concepts.de/wiki/index.php?title=Main_Page</a>.</p>
</div>
</div>
<div class="section" id="id47">
<h2>Conclusions<a class="headerlink" href="#id47" title="Permalink to this headline">¶</a></h2>
<p>CDK-Taverna 2.0 provides an enhanced and matured free open cheminformatics workflow solution for the biosciences. It was successfully applied and tested in academic and industrial environments with data volumes of hundreds of thousands of small molecules. Combined with available workers and workflows from bioinformatics, image analysis or statistics CDK-Taverna supports the construction of complex systems biology oriented workflows for processing diverse sets of biological data.</p>
</div>
<div class="section" id="competing-interests">
<h2>Competing interests<a class="headerlink" href="#competing-interests" title="Permalink to this headline">¶</a></h2>
<p>The authors declare that they have no competing interests.</p>
</div>
<div class="section" id="authors-contributions">
<h2>Authors&#8217; contributions<a class="headerlink" href="#authors-contributions" title="Permalink to this headline">¶</a></h2>
<p>EW initiated the integration of Taverna and CDK and supported deployment and architecture. CS and AZ conceived the project and lead the further development. SN supported the reaction enumeration enhancements. KV provided workers for molecular fragmentation. AT did the majority of CDK-Taverna re-engineering and enhancements and developed the project to its current state. All co-authors contributed to the manuscript. All authors read and approved the final manuscript.</p>
</div>
<div class="section" id="acknowledgements">
<h2>Acknowledgements<a class="headerlink" href="#acknowledgements" title="Permalink to this headline">¶</a></h2>
<p>The authors express their gratitude to the teams and communities of Taverna, CDK and WEKA for creating and developing these open tools.</p>
<table class="docutils footnote" frame="void" id="b1" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id2">[1]</a></td><td>Cheminformatics analysis and learning in a data pipelining environment</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b2" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id3">[2]</a></td><td>Scientific workflows as productivity tools for drug discovery</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b3" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id4">[3]</a></td><td>Taverna/my Grid: Aligning a Workflow System with the Life Sciences Community</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b4" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label">[4]</td><td><em>(<a class="fn-backref" href="#id5">1</a>, <a class="fn-backref" href="#id12">2</a>, <a class="fn-backref" href="#id18">3</a>)</em> CDK-Taverna: an open workflow environment for cheminformatics</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b5" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id6">[5]</a></td><td>Taverna, Reloaded</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b6" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label">[6]</td><td><em>(<a class="fn-backref" href="#id7">1</a>, <a class="fn-backref" href="#id13">2</a>)</em> Taverna 2</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b7" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id8">[7]</a></td><td>Taverna: a tool for the composition and enactment of bioinformatics workflows</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b8" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id9">[8]</a></td><td>The Chemistry Development Kit (CDK): An open-source Java library for chemo- and bioinformatics</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b9" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id10">[9]</a></td><td>Recent Developments of The Chemistry Development Kit (CDK) - An Open-Source Java Library for Chemo- and Bioinformatics</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b10" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id11">[10]</a></td><td>The WEKA Data Mining Software: An Update</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b11" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id14">[11]</a></td><td>The Chemistry Development Kit(CDK)</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b12" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id15">[12]</a></td><td>Waikato Environment for Knowledge Analysis (WEKA)</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b13" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id16">[13]</a></td><td>Apache Maven</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b14" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label">[14]</td><td><em>(<a class="fn-backref" href="#id21">1</a>, <a class="fn-backref" href="#id23">2</a>, <a class="fn-backref" href="#id25">3</a>)</em> Natural Product-likeness Score and Its Application for Prioritization of Compound Libraries</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b15" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id22">[15]</a></td><td>Metabolite-likeness as a criterion in the design and selection of pharmaceutical drug libraries</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b16" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id24">[16]</a></td><td>The Signature Molecular Descriptor. 4. Canonizing Molecules Using Extended Valence Sequences</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b17" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id30">[17]</a></td><td>Maximum likelihood from incomplete data via the EM algorithm</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b18" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id31">[18]</a></td><td>A best possible heuristic for the k-center problem</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b19" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id32">[19]</a></td><td>WEKA API Documentation</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b20" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id33">[20]</a></td><td>Some methods for classification and analysis of multivariate observations</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b21" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id34">[21]</a></td><td>X-means: Extending K-means with an efficient Estimation of the Number of Clusters</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b22" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id35">[22]</a></td><td>NOTITLE!</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b23" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id36">[23]</a></td><td>LIBSVM: a library for support vector machines</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b24" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id37">[24]</a></td><td>Learning with continuous classes</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b25" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id38">[25]</a></td><td>Induction of model trees for predicting continuous classes</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b26" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id39">[26]</a></td><td>Estimating continuous distributions in Bayesian classifiers</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b27" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id40">[27]</a></td><td>NOTITLE!</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b28" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label">[28]</td><td><em>(<a class="fn-backref" href="#id41">1</a>, <a class="fn-backref" href="#id44">2</a>)</em> NOTITLE!</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="b29" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id45">[29]</a></td><td>MediaWiki</td></tr>
</tbody>
</table>
</div>
</div>


          </div>
        </div>
      </div>
        </div>
        <div class="sidebar">
          <h3>Table Of Contents</h3>
          <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-1.html">CurlySMILES: a chemical language to customize and annotate encodings of molecular and nanodevice structures</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-2.html">Prediction of cyclin-dependent kinase 2 inhibitor potency using the fragment molecular orbital method</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-3.html">jCompoundMapper: An open source Java library and command-line tool for chemical fingerprints</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-4.html">PubChem3D: Conformer generation</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-5.html">Modular Chemical Descriptor Language (MCDL): Stereochemical modules</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-6.html">FlaME: Flash Molecular Editor - a 2D structure input tool for the web</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-7.html">Use of structure-activity landscape index curves and curve integrals to evaluate the performance of multiple machine learning prediction models</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-8.html">Confab - Systematic generation of diverse low-energy conformers</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-9.html">PubChem3D: Diversity of shape</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-11.html">Interpreting linear support vector machine models with heat map molecule coloring</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-12.html">Multilevel Parallelization of AutoDock 4.2</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-13.html">PubChem3D: Similar conformers</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-14.html">Analysis of <em>in vitro</em> bioactivity data extracted from drug discovery literature and patents: Ranking 1654 human protein targets by assayed compounds and molecular scaffolds</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-15.html">Resource description framework technologies in chemistry</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-16.html">Semantic Web integration of Cheminformatics resources with the SADI framework</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-17.html">ChemicalTagger: A tool for semantic text-mining in chemistry</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-18.html">AMBIT RESTful web services: an implementation of the OpenTox application programming interface</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-19.html">Linked open drug data for pharmaceutical research and development</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-20.html">Chemical Entity Semantic Specification: Knowledge representation for efficient semantic cheminformatics and facile data integration</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-21.html">Consistent two-dimensional visualization of protein-ligand complex series</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-22.html">Predicting a small molecule-kinase interaction map: A machine learning approach</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-23.html">4D Flexible Atom-Pairs: An efficient probabilistic conformational space comparison for ligand-based virtual screening</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-24.html">Data governance in predictive toxicology: A review</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-25.html">PubChem3D: Shape compatibility filtering using molecular shape quadrupoles</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-26.html">PubChem3D: Biologically relevant 3-D similarity</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-27.html">Theoretical NMR correlations based Structure Discussion</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-28.html">AZOrange - High performance open source machine learning for QSAR modeling in a graphical programming environment</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-29.html">Multiple search methods for similarity-based virtual screening: analysis of search overlap and precision</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-30.html">Structural diversity of biologically interesting datasets: a scaffold analysis approach</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-31.html">Statistical filtering for NMR based structure generation</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-32.html">PubChem3D: a new resource for scientists</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-33.html">Open Babel: An open chemical toolbox</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-34.html">Adventures in public data</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-35.html">Three stories about the conduct of science: Past, future, and present</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-36.html">Openness as infrastructure</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-37.html">Open Data, Open Source and Open Standards in chemistry: The Blue Obelisk five years on</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-38.html">The Quixote project: Collaborative and Open Quantum Chemistry data management in the Internet age</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-39.html">CMLLite: a design philosophy for CML</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-40.html">Mining chemical information from open patents</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-41.html">OSCAR4: a flexible architecture for chemical text-mining</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-42.html">The semantic architecture of the World-Wide Molecular Matrix (WWMM)</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-43.html">The semantics of Chemical Markup Language (CML): dictionaries and conventions</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-44.html">CML: Evolution and design</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-45.html">Ami - The chemist&#8217;s amanuensis</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-46.html">The past, present and future of Scientific discourse</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-47.html">Open Bibliography for Science, Technology, and Medicine</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-48.html">Semantic science and its communication - a personal view</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-49.html">Molecular dynamics simulations and in silico peptide ligand screening of the Elk-1 ETS domain</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-50.html">2D-Qsar for 450 types of amino acid induction peptides with a novel substructure pair descriptor having wider scope</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-51.html">An investigation into pharmaceutically relevant mutagenicity data and the influence on Ames predictive potential</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-52.html">Automated annotation of chemical names in the literature with tunable accuracy</a></li>
<li class="toctree-l1"><a class="reference internal" href="1758-2946-3-53.html">MyChemise: A 2D drawing program that uses morphing for visualisation purposes</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="">New developments on the cheminformatics open workflow environment CDK-Taverna</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#abstract">Abstract</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#background">Background</a></li>
<li class="toctree-l3"><a class="reference internal" href="#results">Results</a></li>
<li class="toctree-l3"><a class="reference internal" href="#conclusions">Conclusions</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#id1">Background</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#implementation">Implementation</a></li>
<li class="toctree-l3"><a class="reference internal" href="#cdk-taverna-2-0-worker-implementation">CDK-Taverna 2.0 worker implementation</a></li>
<li class="toctree-l3"><a class="reference internal" href="#requirements">Requirements</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#results-and-discussion">Results and Discussion</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#advanced-reaction-enumeration">Advanced reaction enumeration</a></li>
<li class="toctree-l3"><a class="reference internal" href="#evaluation-of-small-molecules-for-natural-product-likeness">Evaluation of small molecules for natural product likeness</a></li>
<li class="toctree-l3"><a class="reference internal" href="#clustering-and-machine-learning-applications">Clustering and machine learning applications</a></li>
<li class="toctree-l3"><a class="reference internal" href="#cdk-taverna-2-0-wiki">CDK-Taverna 2.0 Wiki</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#id47">Conclusions</a></li>
<li class="toctree-l2"><a class="reference internal" href="#competing-interests">Competing interests</a></li>
<li class="toctree-l2"><a class="reference internal" href="#authors-contributions">Authors&#8217; contributions</a></li>
<li class="toctree-l2"><a class="reference internal" href="#acknowledgements">Acknowledgements</a></li>
</ul>
</li>
</ul>

          <h3 style="margin-top: 1.5em;">Search</h3>
          <form class="search" action="../search.html" method="get">
            <input type="text" name="q" />
            <input type="submit" value="Go" />
            <input type="hidden" name="check_keywords" value="yes" />
            <input type="hidden" name="area" value="default" />
          </form>
          <p class="searchtip" style="font-size: 90%">
            Enter search terms or a module, class or function name.
          </p>
        </div>
        <div class="clearer"></div>
      </div>
    </div>

    <div class="footer-wrapper">
      <div class="footer">
        <div class="left">
          <a href="1758-2946-3-53.html" title="MyChemise: A 2D drawing program that uses morphing for visualisation purposes"
             >previous</a> |
          <a href="../genindex.html" title="General Index"
             >index</a>
            <br/>
            <a href="../_sources/articles/1758-2946-3-54.txt"
               rel="nofollow">Show Source</a>
        </div>

        <div class="right">
          
    <div class="footer">
        &copy; Copyright 2012, Noel O&#39;Boyle.
      Created using <a href="http://sphinx.pocoo.org/">Sphinx</a> 1.1.3.
    </div>
        </div>
        <div class="clearer"></div>
      </div>
    </div>

  </body>
</html>